// @wedgekit/checkbox/dist/index.cjs
var Y=Object.create,l=Object.defineProperty;var ee=Object.getOwnPropertyDescriptor;var te=Object.getOwnPropertyNames,u=Object.getOwnPropertySymbols,oe=Object.getPrototypeOf,f=Object.prototype.hasOwnProperty,R=Object.prototype.propertyIsEnumerable;var S=(e,t,o)=>t in e?l(e,t,{enumerable:!0,configurable:!0,writable:!0,value:o}):e[t]=o,H=(e,t)=>{for(var o in t||(t={}))f.call(t,o)&&S(e,o,t[o]);if(u)for(var o of u(t))R.call(t,o)&&S(e,o,t[o]);return e};var P=e=>l(e,"__esModule",{value:!0});var N=(e,t)=>{var o={};for(var r in e)f.call(e,r)&&t.indexOf(r)<0&&(o[r]=e[r]);if(e!=null&&u)for(var r of u(e))t.indexOf(r)<0&&R.call(e,r)&&(o[r]=e[r]);return o};var re=(e,t)=>{for(var o in t)l(e,o,{get:t[o],enumerable:!0})},ne=(e,t,o)=>{if(t&&typeof t=="object"||typeof t=="function")for(let r of te(t))!f.call(e,r)&&r!=="default"&&l(e,r,{get:()=>t[r],enumerable:!(o=ee(t,r))||o.enumerable});return e},i=e=>ne(P(l(e!=null?Y(oe(e)):{},"default",e&&e.__esModule&&"default"in e?{get:()=>e.default,enumerable:!0}:{value:e,enumerable:!0})),e);P(exports);re(exports,{StyledLabel:()=>d,default:()=>j});var n=i(require("react")),V=i(require("@wedgekit/primitives")),Z=i(require("@wedgekit/info-button"));var O=i(require("@wedgekit/color")),W=i(require("styled-components")),a=i(require("@wedgekit/primitives")),ie=W.default.label`
  display: block;
  border-radius: 4px;
  margin: -${a.default.base}px;
  padding: ${a.default.base}px;

  ${({disabled:e})=>!e&&`
    &:hover label,
    &:focus-within label {
      color: ${O.default.N600};
    }
  `}
  &:hover {
    cursor: pointer;
  }
  ${a.CursorMixin}
  ${a.OptionMixin}
`,d=ie;var s=i(require("@wedgekit/color")),p=i(require("@wedgekit/primitives")),q=i(require("styled-components")),z=`${p.default.base*5}px`,ae=q.default.input`
  appearance: none;
  background: ${s.default.N050};
  border: 1px solid;
  border-color: ${({status:e="default"})=>p.default.colors.status[e]};
  border-radius: 2px;
  box-sizing: border-box;
  cursor: pointer;
  height: ${z};
  margin: 0;
  width: ${z};
  ${p.CursorMixin};

  &:focus {
    outline: none;
  }

  &:checked:not(:indeterminate) {
    background-image: url("data:image/svg+xml,%3Csvg id='Layer_2' data-name='Layer 2' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cg id='Check'%3E%3Cpath fill='white' d='M22.56,4.27a2.14,2.14,0,0,0-3,0L9,14.83,4.46,10.31a2.14,2.14,0,0,0-3,3l5,5L8.1,20a1.27,1.27,0,0,0,1.77,0L22.56,7.29A2.12,2.12,0,0,0,22.56,4.27Z'/%3E%3C/g%3E%3C/svg%3E%0A");
    background-size: 14px 14px;
    background-repeat: no-repeat;
    background-position: center;
  }

  &:indeterminate {
    background-image: url("data:image/svg+xml,%3Csvg id='Layer_2' data-name='Layer 2' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Crect fill='white' x='0.84' y='10' width='22.33' height='4.09' rx='2.05'/%3E%3C/svg%3E%0A");
    background-size: 14px 14px;
    background-repeat: no-repeat;
    background-position: center;
  }

  &:checked,
  &:indeterminate {
    background-color: ${s.default.B500};
    border: none;
  }

  &:disabled {
    background-color: ${s.default.N200};
    border-color: ${s.default.N200};
  }
`,h=ae;var A=i(require("styled-components")),x=i(require("@wedgekit/primitives")),k=(0,A.default)(x.Label)`
  cursor: pointer;
  ${x.CursorMixin}
`;var B=i(require("styled-components")),le=B.default.div`
  width: fit-content;
`,C=le;var _=i(require("@wedgekit/primitives")),F=i(require("styled-components")),de=(0,F.default)(_.Row)`
  grid-auto-columns: auto;
  grid-template-columns: min-content auto;
  align-items: start;
`,w=de;var se=n.default.forwardRef((e,t)=>{let T=e,{active:o,className:r,disabled:c,id:v,indeterminate:y,info:L,label:D,labelHidden:G,name:E,onChange:J,required:M,status:K,tabIndex:$,value:Q}=T,U=N(T,["active","className","disabled","id","indeterminate","info","label","labelHidden","name","onChange","required","status","tabIndex","value"]);E&&console.warn("The `name` prop for Checkbox has been deprecated. Functionality has been migrated to the `id` prop instead");let m=(0,n.useRef)(null);(0,n.useImperativeHandle)(t,()=>m.current),(0,n.useEffect)(()=>{m.current&&(m.current.indeterminate=!!y)},[y]);let g=!!Q,X=b=>{J(!g,v||"",b)},I=(0,n.useRef)(null);return n.default.createElement(C,{className:r},n.default.createElement(d,{active:o,disabled:c,ref:I,onClick:b=>{b.target===I.current&&b.stopPropagation()}},n.default.createElement(w,null,n.default.createElement(h,H({"aria-selected":g,checked:g,disabled:c,id:v,name:E,onChange:X,ref:m,required:M,status:K,type:"checkbox",tabIndex:c?V.default.tabIndex.disabled:$},U)),!G&&n.default.createElement(k,{as:"div",disabled:c,required:M},D),L&&n.default.createElement(Z.default,{info:L,tabIndex:$}))))}),j=se;
