// @wedgekit/palette/dist/index.js
const colors = {
    /**
     * 700 (darkest) level of dark blue
     */
    D700: '#002131',
    /**
     * 600 level of dark blue
     */
    D600: '#00354E',
    /**
     * 500 level of dark blue
     */
    D500: '#00486B',
    /**
     * 400 level of dark blue
     */
    D400: '#457993',
    /**
     * 300 level of dark blue
     */
    D300: '#8BABBB',
    /**
     * 200 level of dark blue
     */
    D200: '#D0DDE4',
    /**
     * 100 (lightest) level of dark blue
     */
    D100: '#E7EEF1',
    /**
     * 700 (darkest) level of blue
     */
    B700: '#00365A',
    /**
     * 600 level of blue
     */
    B600: '#00568F',
    /**
     * 500 level of blue
     */
    B500: '#0075C4',
    /**
     * 400 level of blue
     */
    B400: '#459AD4',
    /**
     * 300 level of blue
     */
    B300: '#8BC0E4',
    /**
     * 200 level of blue
     */
    B200: '#D0E5F4',
    /**
     * 100 (lightest) level of blue
     */
    B100: '#E7F2F9',
    /**
     * 700 (darkest) level of green
     */
    G700: '#126240',
    /**
     * 600 level of green
     */
    G600: '#1D9C66',
    /**
     * 500 level of green
     */
    G500: '#27D68B',
    /**
     * 400 level of green
     */
    G400: '#61E1AA',
    /**
     * 300 level of green
     */
    G300: '#9CECCA',
    /**
     * 200 level of green
     */
    G200: '#D7F7E9',
    /**
     * 100 (lightest) level of green
     */
    G100: '#EEFBF6',
    /**
     * 700 (darkest) level of red
     */
    R700: '#892615',
    /**
     * 600 level of red
     */
    R600: '#B0311B',
    /**
     * 500 level of red
     */
    R500: '#D63B20',
    /**
     * 400 level of red
     */
    R400: '#DD5E48',
    /**
     * 300 level of red
     */
    R300: '#E48271',
    /**
     * 200 level of red
     */
    R200: '#F0B7AD',
    /**
     * 100 (lightest) level of red
     */
    R100: '#FBEDEA',
    /**
     * 700 (darkest) level of yellow
     */
    Y700: '#996C0D',
    /**
     * 600 level of yellow
     */
    Y600: '#C58B10',
    /**
     * 500 level of yellow
     */
    Y500: '#F0A913',
    /**
     * 400 level of yellow
     */
    Y400: '#F4C053',
    /**
     * 300 level of yellow
     */
    Y300: '#F6D07E',
    /**
     * 200 level of yellow
     */
    Y200: '#F9DFA9',
    /**
     * 100 (lightest) level of yellow
     */
    Y100: '#FCEFD4',
    /**
     * 700 (darkest) level of greyscale
     */
    N700: '#353636',
    /**
     * 600 level of greyscale
     */
    N600: '#555656',
    /**
     * 500 level of greyscale
     */
    N500: '#6B6D6D',
    /**
     * 400 level of greyscale
     */
    N400: '#999B9B',
    /**
     * 300 level of greyscale
     */
    N300: '#BFC0C0',
    /**
     * 200 level of greyscale
     */
    N200: '#D9D9D9',
    /**
     * 100 level of greyscale
     */
    N100: '#F2F2F2',
    /**
     * 50 (lightest) level of greyscale
     */
    N050: '#FFFFFF',
    /**
     * tag color
     */
    T100: '#005BA6',
    /**
     * tag color
     */
    T110: '#00AEEF',
    /**
     * tag color
     */
    T200: '#00950B',
    /**
     * tag color
     */
    T210: '#72EE59',
    /**
     * tag color
     */
    T300: '#FFFF00',
    /**
     * tag color
     */
    T310: '#F7941D',
    /**
     * tag color
     */
    T400: '#FF0000',
    /**
     * tag color
     */
    T410: '#C12275',
    /**
     * tag color
     */
    T420: '#FA8CBF',
    /**
     * tag color
     */
    T500: '#762D91',
    /**
     * tag color
     */
    T600: '#754C24',
    /**
     * tag color
     */
    T700: '#000000',
    /**
     * START Denim color family
     */
    /* Darkest denim */
    DE700: '#00486B',
    DE650: '#336D89',
    DE400: '#8DADBD',
    DE100: '#D4DFE5',
    DE050: '#EBF0F3',
    /**
     * END Denim color family
     */
    /**
     * START Graphite color family
     */
    GR700: '#353636',
    GR650: '#5D5E5E',
    GR400: '#A5A5A5',
    GR100: '#DCDDDD',
    GR050: '#EFEFEF',
    /**
     * END Graphite color family
     */
    /**
     * START Sage color family
     */
    SA700: '#006B58',
    SA650: '#338979',
    SA400: '#8DBDB4',
    SA100: '#D4E5E3',
    SA050: '#EBF3F2',
    /**
     * END Sage color family
     */
    /**
     * START Terra Cotta color family
     */
    TC700: '#B44100',
    TC650: '#C36733',
    TC450: '#FF9458',
    TC400: '#DDAA8D',
    TC100: '#F2DFD4',
    TC050: '#F9F0EB',
    /**
     * END Terra Cotta color family
     */
    /**
     * START Violet color family
     */
    VI700: '#4E008B',
    VI650: '#7233A2',
    VI400: '#B899D1',
    VI100: '#E0D4EB',
    VI050: '#F1EBF6',
    /**
     * END Violet color family
     */
    /**
     * START Azure color family
     */
    AZ700: '#003DB4',
    AZ650: '#3364C3',
    AZ450: '#689BFF',
    AZ400: '#8DA8DD',
    AZ100: '#D4DDF2',
    AZ050: '#EBEFF9',
    /**
     * END Azure color family
     */
    /**
     * START Carmine color family
     */
    CA700: '#960018',
    CA650: '#BF0020',
    CA400: '#FF7089',
    CA150: '#FFC3CE',
    /**
     * END Carmine color family
     */
};
export default colors;
