// @wedgekit/indicators/dist/index.cjs
var L=Object.create,l=Object.defineProperty,W=Object.defineProperties,C=Object.getOwnPropertyDescriptor,T=Object.getOwnPropertyDescriptors,Y=Object.getOwnPropertyNames,s=Object.getOwnPropertySymbols,N=Object.getPrototypeOf,y=Object.prototype.hasOwnProperty,S=Object.prototype.propertyIsEnumerable;var w=(o,t,e)=>t in o?l(o,t,{enumerable:!0,configurable:!0,writable:!0,value:e}):o[t]=e,i=(o,t)=>{for(var e in t||(t={}))y.call(t,e)&&w(o,e,t[e]);if(s)for(var e of s(t))S.call(t,e)&&w(o,e,t[e]);return o},d=(o,t)=>W(o,T(t)),P=o=>l(o,"__esModule",{value:!0});var $=(o,t)=>{var e={};for(var r in o)y.call(o,r)&&t.indexOf(r)<0&&(e[r]=o[r]);if(o!=null&&s)for(var r of s(o))t.indexOf(r)<0&&S.call(o,r)&&(e[r]=o[r]);return e};var D=(o,t)=>{for(var e in t)l(o,e,{get:t[e],enumerable:!0})},O=(o,t,e)=>{if(t&&typeof t=="object"||typeof t=="function")for(let r of Y(t))!y.call(o,r)&&r!=="default"&&l(o,r,{get:()=>t[r],enumerable:!(e=C(t,r))||e.enumerable});return o},a=o=>O(P(l(o!=null?L(N(o)):{},"default",o&&o.__esModule&&"default"in o?{get:()=>o.default,enumerable:!0}:{value:o,enumerable:!0})),o);P(exports);D(exports,{Loading:()=>b,Submit:()=>h});var m=a(require("react"));var v=a(require("styled-components")),g=a(require("@wedgekit/color")),u=a(require("@wedgekit/primitives")),_=v.default.div`
  height: 100%;
  width: ${({size:o})=>o?`${o/5}px`:(0,u.getBasePx)(1)};
  margin: 0 ${({size:o})=>o?`${o/20}px`:(0,u.getBasePx)(.25)};
  display: inline-block;
  animation: sk-stretchdelay 1.2s infinite ease-in-out;
  border-radius: 2px;
  background: ${({color:o})=>o?g.default[o]:g.default.D500};
  animation-delay: ${({delay:o})=>o}s;

  @keyframes sk-stretchdelay {
    0%,
    50%,
    100% {
      transform: scaleY(0.4);
      -webkit-transform: scaleY(0.4);
      opacity: 0;
    }

    20% {
      transform: scaleY(1);
      -webkit-transform: scaleY(1);
      opacity: 1;
    }
  }
`,n=_;var B=a(require("styled-components")),I=a(require("@wedgekit/primitives")),j=B.default.div`
  height: ${({size:o})=>o?`${o}px`:(0,I.getBasePx)(4)};
  text-align: center;
  position: relative;
  width: max-content;

  &::after {
    position: absolute;
    left: 50%;
    width: 200px;
    margin-left: -100px;
    display: block;
    content: attr(data-label);
  }
`,x=j;var q=e=>{var r=e,{className:o}=r,t=$(r,["className"]);return m.default.createElement(x,i({className:o},t),m.default.createElement(n,d(i({},t),{delay:-1.1})),m.default.createElement(n,d(i({},t),{delay:-1})),m.default.createElement(n,d(i({},t),{delay:-.9})),m.default.createElement(n,d(i({},t),{delay:-.8})),m.default.createElement(n,d(i({},t),{delay:-.6})))},b=q;var f=a(require("react")),c=a(require("@wedgekit/icons"));var p=a(require("styled-components")),z=p.keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`,A=p.default.div`
  animation: ${z} 1s linear infinite;
  width: min-content;
`,k=A;var E=o=>f.default.createElement(k,null,f.default.createElement(c.IconWidth,{iconWidth:o.size||20},f.default.createElement(c.default,i({},o),"loading_circle"))),h=E;
