// @wedgekit/layout/dist/index.cjs
var fe=Object.create,w=Object.defineProperty,ge=Object.defineProperties,we=Object.getOwnPropertyDescriptor,ye=Object.getOwnPropertyDescriptors,Pe=Object.getOwnPropertyNames,x=Object.getOwnPropertySymbols,he=Object.getPrototypeOf,D=Object.prototype.hasOwnProperty,z=Object.prototype.propertyIsEnumerable;var k=(e,r,t)=>r in e?w(e,r,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[r]=t,d=(e,r)=>{for(var t in r||(r={}))D.call(r,t)&&k(e,t,r[t]);if(x)for(var t of x(r))z.call(r,t)&&k(e,t,r[t]);return e},_=(e,r)=>ge(e,ye(r)),H=e=>w(e,"__esModule",{value:!0});var T=(e,r)=>{var t={};for(var o in e)D.call(e,o)&&r.indexOf(o)<0&&(t[o]=e[o]);if(e!=null&&x)for(var o of x(e))r.indexOf(o)<0&&z.call(e,o)&&(t[o]=e[o]);return t};var xe=(e,r)=>{for(var t in r)w(e,t,{get:r[t],enumerable:!0})},Se=(e,r,t)=>{if(r&&typeof r=="object"||typeof r=="function")for(let o of Pe(r))!D.call(e,o)&&o!=="default"&&w(e,o,{get:()=>r[o],enumerable:!(t=we(r,o))||t.enumerable});return e},n=e=>Se(H(w(e!=null?fe(he(e)):{},"default",e&&e.__esModule&&"default"in e?{get:()=>e.default,enumerable:!0}:{value:e,enumerable:!0})),e);H(exports);xe(exports,{default:()=>Be});var i=n(require("react"));var U=n(require("styled-components")),be=U.default.div`
  height: 100%;
  overflow: hidden;
  display: grid;
  grid-template-rows: minmax(0, max-content) minmax(0, 1fr);
`,L=be;var q=n(require("styled-components")),Me=q.default.div`
  flex: 1;
  min-height: min-content;
  display: flex;
  z-index: 1;
  width: 100%;
  overflow: hidden;
`,C=Me;var F=n(require("styled-components")),De=F.default.div`
  position: relative;
  height: 100%;
  overflow-y: auto;
`,$=De;var y=n(require("styled-components")),B=n(require("@wedgekit/color")),Le=y.keyframes`
  0% {
    right: -25%;
    @media screen and (max-width: 1024px) {
      right: -50%;
    }
  }
  
  100% {
    right: 0;
  }
`,m=y.default.div`
  background: ${B.default.N050};
  width: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  height: 100%;
  box-sizing: border-box;
`,p=(0,y.default)(m)`
  position: absolute;
  width: 25%;
  right: 0;
  animation: ${Le} 0.2s ease-in;
  z-index: 1;

  @media screen and (max-width: 1280px) {
    width: 50%;
  }
`;var J=n(require("styled-components")),Ce=J.default.div`
  position: relative;
`,N=Ce;var S=n(require("styled-components")),b=n(require("@wedgekit/color")),$e=S.keyframes`
  0% {
    background: ${b.default.D700}00;
  }
  
  100% {
    background: ${b.default.D700}aa;
  }
`,u=S.default.div`
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: ${b.default.D700}aa;
  animation: ${$e} 0.1s ease-in;
  z-index: 1;
`;var K=n(require("styled-components")),Ne=K.default.div`
  z-index: 2;
`,v=Ne;var Q=n(require("styled-components")),ve=Q.default.main`
  width: 100%;
`,R=ve;var V=n(require("styled-components")),Re=V.default.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  position: relative;
  z-index: 1;
  max-width: 100%;
  box-sizing: border-box;
  overflow-x: hidden;

  /*
    Explicitly set overflow-y to auto to fix a bug in Edge:
    https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/10795984/

    Other browsers compute overflow-y to auto when overflow-x is set to hidden.
  */
  overflow-y: auto;
`,A=Re;var Y=n(require("styled-components")),Ae=Y.default.div`
  z-index: 3;
  position: absolute;
  width: 95%;
  max-width: 768px;
  top: 0;
  left: 50%;

  transform: translate(-50%, 0);
`,G=Ae;var j=n(require("styled-components")),Ge=j.default.div`
  display: grid;
  height: 100%;
  width: 100%;
  transition: grid-template-columns 0.2s ease-out;
`,P=(0,j.default)(Ge)`
  grid-template-rows: 100%;
  grid-template-columns: ${e=>{var t;let r=(t=e.drawerWidth)!=null?t:"25%";return e.drawerLeft?e.resourceMounted?`${r} 1fr`:"1fr":e.resourceMounted?`1fr ${r}`:"1fr 0"}};

  @media screen and (max-width: 1024px) {
    grid-template-columns: ${e=>{var t;let r=(t=e.drawerWidth)!=null?t:"40%";return e.drawerLeft?e.resourceMounted?`${r} 1fr`:"1fr":e.resourceMounted?`1fr ${r}`:"1fr 0"}};
  }
`;var je=({content:e,resource:r,resourceMounted:t,drawerLeft:o,drawerWidth:c})=>i.createElement(P,{resourceMounted:t,drawerLeft:o,drawerWidth:c},t&&o&&i.createElement(m,null,r),e,t&&!o&&i.createElement(m,null,r)),Oe=({content:e,resource:r,resourceMounted:t,drawerLeft:o})=>i.createElement(N,null,t&&o&&i.createElement(u,null,i.createElement(p,null,r)),e,t&&!o&&i.createElement(u,null,i.createElement(p,null,r)));function Z(e){let{resource:r,resourceMounted:t,resourceType:o,drawerLeft:c,drawerWidth:h}=e,f={drawer:je,"drawer-mobile":Oe},g=i.createElement($,{id:"main-wrapper"},i.createElement(A,null,i.createElement(R,{id:"main"},i.createElement(C,null,e.children({resourceMounted:t})))),i.createElement(G,null,e.notifications)),a=o&&f[o]?f[o]({content:g,resource:r,resourceMounted:t,drawerLeft:c,drawerWidth:h}):g;return i.createElement(L,null,i.createElement(v,null,e.header),a)}Z.defaultProps={header:null,notifications:null,resourceType:"",drawerLeft:!1};var ee=Z;var re=ee;var ie=n(require("react")),le=n(require("styled-components")),de=n(require("@wedgekit/primitives"));var Xe="Default row settings must be supplied to the LayoutNext component if breakpoints are being used.",Ee="Default column settings must be supplied to the LayoutNext component if breakpoints are being used.",Ie="Area settings must be supplied to the LayoutNext component.",We="Area settings must have a consistent number of columns across each row provided per setting.",te=["rowsLg","rowsMd","rowsSm"],oe=["columnsLg","columnsMd","columnsSm"],O=["areasLg","areasMd","areasSm"],ze={areas:"columns",areasLg:"columnsLg",areasMd:"columnsMd",areasSm:"columnsSm"},ae=e=>e.map(r=>typeof r=="number"?`minmax(0, ${r}fr)`:r).join(" "),X=e=>e.map(r=>typeof r=="number"?`minmax(0, ${r}fr)`:r).join(" "),se=e=>e.map(r=>`"${r}"`).join(`
`),ne=e=>{var c,h,f,g;let r={};if(!e.areas)throw new Error(Ie);["areas",...O].forEach(a=>{if(e[a]){let s={};if(e[a].forEach(M=>{s[M.split(" ").length]=!0}),Object.keys(s).length>1)throw new Error(We)}});let t=Object.keys(e).some(a=>{var s;return te.includes(a)&&((s=e[a])==null?void 0:s.length)}),o=Object.keys(e).some(a=>{var s;return oe.includes(a)&&((s=e[a])==null?void 0:s.length)});if(t&&!((c=e.rows)==null?void 0:c.length))throw new Error(Xe);if(o&&!((h=e.columns)==null?void 0:h.length))throw new Error(Ee);return((f=e.rows)==null?void 0:f.length)?(r.rows=ae(e.rows),te.forEach(a=>{var s;((s=e[a])==null?void 0:s.length)&&(r[a]=ae(e[a]))})):r.rows="auto",((g=e.columns)==null?void 0:g.length)?(r.columns=X(e.columns),oe.forEach(a=>{var s;((s=e[a])==null?void 0:s.length)&&(r[a]=X(e[a]))})):["areas",...O].forEach(a=>{var s;if((s=e[a])==null?void 0:s.length){let M=e[a][0].split(" ").length;r[ze[a]]=X(Array(M).fill(void 0).map(()=>1))}}),r.areas=se(e.areas),O.forEach(a=>{var s;((s=e[a])==null?void 0:s.length)&&(r[a]=se(e[a]))}),r};var me=e=>e.noGap?"0px":`${e.multiplier*de.default.base}px`,ke=le.default.div`
  display: grid;
  grid-template-rows: ${e=>e.rowsXl||e.rows};
  grid-template-columns: ${e=>e.columnsXl||e.columns};
  grid-template-areas: ${e=>e.areasXl||e.areas};
  grid-column-gap: ${e=>me(e)};
  grid-row-gap: ${e=>me(e)};
  width: 100%;
  justify-content: ${e=>e.justify};
  justify-items: ${e=>e.justifyItems};
  align-items: ${e=>e.align};
  place-items: ${e=>e.place};

  @media screen and (max-width: 1200px) {
    grid-template-rows: ${e=>e.rowsLg};
    grid-template-columns: ${e=>e.columnsLg};
    grid-template-areas: ${e=>e.areasLg};
  }

  @media screen and (max-width: 992px) {
    grid-template-rows: ${e=>e.rowsMd};
    grid-template-columns: ${e=>e.columnsMd};
    grid-template-areas: ${e=>e.areasMd};
  }

  @media screen and (max-width: 768px) {
    grid-template-rows: ${e=>e.rowsSm};
    grid-template-columns: ${e=>e.columnsSm};
    grid-template-areas: ${e=>e.areasSm};
  }

  @media screen and (max-width: 576px) {
    grid-template-rows: ${e=>e.rowsXs};
    grid-template-columns: ${e=>e.columnsXs};
    grid-template-areas: ${e=>e.areasXs};
  }
`,ce=e=>{let{className:r}=e,t=ne(e),o=d({},t);return e["data-cy"]&&(o["data-cy"]=e["data-cy"]),ie.default.createElement(ke,_(d({},o),{className:r,multiplier:parseInt((e.multiplier||1).toString(),10),align:e.align,justify:e.justify,noGap:e.noGap,place:e.place}),e.children)};ce.defaultProps={areasLg:null,areasMd:null,areasSm:null,columnsLg:null,columnsMd:null,columnsSm:null,rowsLg:null,rowsMd:null,rowsSm:null,justify:null,align:null};var E=ce;var pe=n(require("react")),ue=n(require("styled-components")),_e=ue.default.div`
  grid-area: ${e=>e.area};
  align-self: ${e=>e.align};
  justify-self: ${e=>e.justify};
  place-self: ${e=>e.place};
`,He=t=>{var o=t,{children:e}=o,r=T(o,["children"]);return pe.default.createElement(_e,d({},r),e)},I=He;var l=n(require("react"));var Te=({resourceMounted:e,children:r,resource:t})=>l.default.createElement(P,{resourceMounted:e},r({resourceMounted:e}),e&&l.default.createElement(m,null,t)),Ue=({resourceMounted:e,children:r,resource:t})=>l.default.createElement(l.default.Fragment,null,r({resourceMounted:e}),e&&l.default.createElement(u,null,l.default.createElement(p,null,t))),qe=e=>(console.warn("Layout.Page is deprecated and will be removed in v1.0.0. Please switch to using resources in Layout.App"),e.type==="drawer"?l.default.createElement(Te,d({},e)):e.type==="drawer-mobile"?l.default.createElement(Ue,d({},e)):e.children()),W=qe;var Fe={App:re,Grid:E,Page:W,Section:I},Be=Fe;
