// @wedgekit/primitives/dist/index.cjs
var Fo=Object.create,x=Object.defineProperty,Co=Object.defineProperties,No=Object.getOwnPropertyDescriptor,Oo=Object.getOwnPropertyDescriptors,Lo=Object.getOwnPropertyNames,S=Object.getOwnPropertySymbols,Ro=Object.getPrototypeOf,L=Object.prototype.hasOwnProperty,eo=Object.prototype.propertyIsEnumerable;var to=(o,t,n)=>t in o?x(o,t,{enumerable:!0,configurable:!0,writable:!0,value:n}):o[t]=n,p=(o,t)=>{for(var n in t||(t={}))L.call(t,n)&&to(o,n,t[n]);if(S)for(var n of S(t))eo.call(t,n)&&to(o,n,t[n]);return o},R=(o,t)=>Co(o,Oo(t)),ro=o=>x(o,"__esModule",{value:!0});var B=(o,t)=>{var n={};for(var i in o)L.call(o,i)&&t.indexOf(i)<0&&(n[i]=o[i]);if(o!=null&&S)for(var i of S(o))t.indexOf(i)<0&&eo.call(o,i)&&(n[i]=o[i]);return n};var zo=(o,t)=>{for(var n in t)x(o,n,{get:t[n],enumerable:!0})},Ho=(o,t,n)=>{if(t&&typeof t=="object"||typeof t=="function")for(let i of Lo(t))!L.call(o,i)&&i!=="default"&&x(o,i,{get:()=>t[i],enumerable:!(n=No(t,i))||n.enumerable});return o},r=o=>Ho(ro(x(o!=null?Fo(Ro(o)):{},"default",o&&o.__esModule&&"default"in o?{get:()=>o.default,enumerable:!0}:{value:o,enumerable:!0})),o);ro(exports);zo(exports,{BackgroundMixin:()=>w,CursorMixin:()=>M,FieldStatusWrapper:()=>H,Fieldset:()=>z,FocusIndicator:()=>U,Gutter:()=>G,InputBorderMixin:()=>v,InputFocusMixin:()=>P,InputMixin:()=>K,Label:()=>A,Legend:()=>D,Link:()=>q,OptionMixin:()=>Q,PopoverShadowMixin:()=>V,Row:()=>E,ShadowMixin:()=>Y,Text:()=>X,Title:()=>J,UpdatedInputMixin:()=>Z,default:()=>ce,getBasePx:()=>g,withUpDown:()=>To});var I=r(require("styled-components")),b=r(require("@wedgekit/color")),d=4,g=(o,t=0)=>`${d*o+t}px`,Uo={animation:{transform:I.css`
      transition: transform 0.2s ease-out;
    `,boxShadow:I.css`
      transition: box-shadow 0.2s;
    `,border:I.css`
      transition: border 0.2s;
    `},base:d,border:{width:{standard:1,selected:1.5},radius:4},colors:{status:{default:b.default.N500,error:b.default.R400,pending:b.default.B500,success:b.default.G500}},font:{code:"Courier New"},fontFamily:"'Open Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif",fontSize:{label:d*4,link:d*4,body:d*4,bodySmall:d*2.5,input:d*4,tooltip:d*2.5,h1:d*6,h2:d*5,h3:d*4.5,h4:d*4,h5:d*3.5,h6:d*3},fontWeight:{title:700,normal:400,link:600},widths:{toast:550,multiselectCaret:13.5},zindex:{modal:1050,header:1030},cursor:{disabled:"not-allowed",readOnly:"text"},tabIndex:{disabled:-1}},e=Uo;var no=r(require("styled-components")),Go=no.default.fieldset`
  padding: 0;
  margin: 0;
  border: 0;
`,z=Go;var io=r(require("styled-components")),$=r(require("@wedgekit/color")),Wo=(o,t)=>o.disabled?$.default.N400:o.status==="error"||o.invalid?$.default.R500:o.readOnly?$.default.N600:$.default[t],Do=io.default.div`
  width: ${o=>o.fullLength&&"100%"};
  label {
    color: ${o=>Wo(o,"N600")};
  }
`,H=Do;var ao=r(require("styled-components")),y=r(require("@wedgekit/color")),so=(o,t)=>o.disabled?y.default.N400:o.status==="error"||o.invalid?y.default.R500:o.readOnly?y.default.N600:y.default[t],Ao=ao.default.div`
  label {
    color: ${o=>so(o,"N600")};
  }

  &:focus-within label {
    color: ${o=>so(o,"B500")};
  }
`,U=Ao;var T=r(require("react")),lo=r(require("styled-components"));var qo=lo.default.div`
  margin: ${e.base}px 0;
  overflow: hidden;
  display: block;
`,Eo=o=>((0,T.useEffect)(()=>{console.log("Gutter is deprecated and will be removed in @wedgekit/primitives@3.x.x")}),T.default.createElement(qo,p({},o))),G=Eo;var h=r(require("styled-components")),W=r(require("@wedgekit/color"));var Xo=o=>o.required?'"(required)"':o.readOnly?'"(read only)"':"",po=h.css`
  display: block;
  font-size: ${e.fontSize.label}px;
  color: ${o=>o.disabled?W.default.N400:W.default.N600};
  font-weight: 600;
  line-height: 18px;
  font-family: ${e.fontFamily};
  cursor: text;

  &::after {
    content: ${o=>Xo(o)};
    position: relative;
    font-style: italic;
    font-weight: normal;
    font-size: ${e.fontSize.label}px;
    font-family: ${e.fontFamily};
    padding-left: ${e.base}px;
  }
`,jo=h.default.label`
  ${po}
`,D=h.default.legend`
  padding-left: 0;
  ${po}
`,A=jo;var uo=r(require("styled-components")),s=r(require("@wedgekit/color"));var Jo=uo.default.a`
  color: ${({dark:o})=>o?s.default.B300:s.default.B500};
  font-size: ${e.fontSize.link}px;
  font-weight: ${e.fontWeight.link};
  padding: ${({inline:o})=>o?`${e.base*.5}px ${e.base}px`:`${e.base}px`};
  margin: ${({inline:o})=>o?`${e.base*-.5}px ${e.base*-1}px`:"0px"};
  background: transparent;
  width: max-content;
  border-radius: 2px;
  font-family: ${e.fontFamily};
  text-decoration: ${({inline:o,underline:t})=>o||t?"underline":"none"};
  cursor: pointer;

  &:hover,
  &:focus {
    background: ${({dark:o})=>o?(0,s.hexToRGBA)(s.default.B600,.8):s.default.N200};
    color: ${({dark:o})=>o?s.default.B100:s.default.D700};
    outline: none;
  }
`,q=Jo;var mo=r(require("styled-components"));var Ko=mo.default.div`
  width: 100%;
  display: grid;
  grid-auto-columns: max-content;
  grid-template-rows: auto;
  grid-auto-flow: column;
  grid-column-gap: ${e.base*2}px;
  align-items: ${o=>o.center?"center":"end"};
`,E=Ko;var fo=r(require("styled-components")),co=r(require("@wedgekit/color"));var Qo=fo.default.p`
  font-size: ${o=>e.fontSize[o.small?"bodySmall":"body"]}px;
  color: ${co.default.N600};
  font-family: ${e.fontFamily};
  white-space: pre-wrap;
  margin: 0;
`,X=Qo;var a=r(require("styled-components")),F=r(require("react")),j=r(require("@wedgekit/color"));var f=a.css`
  margin: 0;
  padding: 0;
  font-weight: ${e.fontWeight.title};
  color: ${o=>o.white?j.default.N050:j.default.D500};
  font-size: ${o=>e.fontSize[`h${o.level}`]}px;
  font-family: ${e.fontFamily};
  line-height: 1.3;
`,Vo=a.default.div`
  ${f}
`,Yo=a.default.h1`
  ${f}
`,Zo=a.default.h2`
  ${f}
`,_o=a.default.h3`
  ${f}
`,oe=a.default.h4`
  ${f}
`,ee=a.default.h5`
  ${f}
`,te=a.default.h6`
  ${f}
`,re=(o,t)=>{if(!t)return Vo;switch(t){case 1:return Yo;case 2:return Zo;case 3:return _o;case 4:return oe;case 5:return ee;default:return te}},ne=n=>{var i=n,{forwardedRef:o}=i,t=B(i,["forwardedRef"]);let _=re(t.level,t.elementLevel);return F.default.createElement(_,R(p({},t),{ref:o}),t.children)},xo=F.default.forwardRef((o,t)=>F.default.createElement(ne,R(p({},o),{forwardedRef:t})));xo.displayName="Title";var J=xo;var bo=r(require("styled-components")),C=r(require("@wedgekit/color")),ie=(o,t)=>o.disabled?C.default.N200:o.readOnly?C.default.N100:C.default[t],de=bo.css`
  background-color: ${o=>ie(o,"N050")};
`,w=de;var go=r(require("styled-components"));var $o=o=>o.disabled?e.cursor.disabled:o.readOnly?e.cursor.readOnly:"",ae=go.css`
  cursor: ${o=>$o(o)};

  &:hover {
    cursor: ${o=>$o(o)};
  }
`,M=ae;var yo=r(require("styled-components")),l=r(require("@wedgekit/color"));var N=(o,t)=>o.invalid||o.status==="error"?[`${e.border.width.standard}px solid ${l.default.R500}`,`inset 0 0 0 .5px ${l.default.R500}`]:o.wrapsGroup?[]:o.disabled?[`${e.border.width.standard}px solid ${l.default.N200}`]:t?o.readOnly?[`${e.border.width.standard}px solid ${l.default.B400}`,`inset 0 0 0 .5px ${l.default.B400}`]:[`${e.border.width.standard}px solid ${l.default.B500}`,`inset 0 0 0 .5px ${l.default.B500}`]:[`${e.border.width.standard}px solid ${l.default.N400}`],v=yo.css`
  border: ${o=>N(o,!1)[0]};
  box-shadow: ${o=>N(o,!1)[1]};

  &:focus {
    border: ${o=>N(o,!0)[0]};
    box-shadow: ${o=>N(o,!0)[1]};
  }
`;var ho=r(require("styled-components")),P=ho.css`
  border-width: 2px;
  outline: none;
`;var wo=r(require("styled-components")),u=r(require("@wedgekit/color"));var se=wo.css`
  appearance: none;
  font-family: ${e.fontFamily};
  background: ${u.default.N050};
  font-size: ${e.fontSize.input}px;
  padding: ${e.base*2}px;
  border-radius: 2px;
  border: ${o=>o.invalid?`1px solid ${u.default.R500}`:`1px solid ${e.colors.status.default}`};
  width: ${o=>o.fullWidth&&"100%"};
  margin: ${o=>o.labelHidden||o.labelInside?0:e.base}px 0 0;
  box-sizing: border-box;
  background: ${o=>o.disabled&&u.default.N200};
  border-color: ${o=>o.disabled&&u.default.N200};
  box-shadow: none;

  &:focus {
    ${P}
    border-color: ${o=>o.invalid?u.default.R500:u.default.B500};
    padding: ${e.base*2-1}px;
    border-color: ${o=>o.disabled&&u.default.N200};
  }
`,K=se;var Mo=r(require("styled-components")),m=r(require("@wedgekit/color")),le=Mo.css`
  background: ${({active:o})=>o&&(0,m.hexToRGBA)(m.default.B500,.1)};
  &:hover {
    background: ${({disabled:o})=>!o&&(0,m.hexToRGBA)(m.default.B500,.1)};
  }

  &:focus-within {
    background: ${({disabled:o})=>!o&&(0,m.hexToRGBA)(m.default.B500,.1)};
  }
`,Q=le;var vo=r(require("styled-components")),Po=r(require("@wedgekit/color")),pe=vo.css`
  box-shadow: 0px 1px 4px ${Po.default.N700}20;
`,V=pe;var ko=r(require("styled-components")),c=r(require("@wedgekit/color"));var So=(o,t)=>(console.log("ShadowMixin is deprecated and will be removed in @wedgekit/primitives@3.x.x"),o.wrapsGroup?o.status==="error"?`inset 0 0 0 ${e.border.width.standard}px ${c.default.R500}`:"":o.invalid||o.status==="error"?`inset 0 0 0 ${e.border.width.selected}px ${c.default.R500}`:o.disabled?`inset 0 0 0 ${e.border.width.standard}px ${c.default.N200}`:t?o.readOnly?`inset 0 0 0 ${e.border.width.selected}px ${c.default.B400}`:`inset 0 0 0 ${e.border.width.selected}px ${c.default.B500}`:`inset 0 0 0 ${e.border.width.standard}px ${c.default.N400}`),Y=ko.css`
  box-shadow: ${o=>So(o,!1)};

  &:focus {
    box-shadow: ${o=>So(o,!0)};
  }
`;var k=r(require("@wedgekit/color")),Bo=r(require("styled-components"));var ue=Bo.css`
  appearance: none;
  border-radius: ${e.border.radius}px;
  ${e.animation.border}
  ${e.animation.boxShadow}
  font-family: ${e.fontFamily};
  font-size: ${e.fontSize.input}px;
  line-height: 1.5rem;
  color: ${({disabled:o})=>o?k.default.N400:k.default.N600};
  -webkit-text-fill-color: ${({disabled:o})=>o&&k.default.N400};
  width: ${o=>o.fullWidth&&"100%"};
  height: 40px;
  box-sizing: border-box;
  border: none;
  outline: none;
  padding: ${g(3,-1)};
  margin-top: ${o=>o.labelHidden||o.labelInline?"0px":g(3)};

  &::placeholder {
    color: ${k.default.N400};
  }

  ${M};
  ${v};
  ${w};
`,Z=ue;var O=r(require("react")),Io=r(require("styled-components"));var me=Io.default.div`
  svg {
    transform: ${o=>o.up?"rotateX(180deg)":"rotateX(0deg)"};
    ${e.animation.transform}
  }
`,fe=(o,{icon:t})=>_=>{var oo=_,{direction:n}=oo,i=B(oo,["direction"]);return t?O.default.createElement(me,{up:n==="up"},O.default.createElement(o,p({},i))):O.default.createElement(o,p({},i))},To=fe;var ce=e;
