// @wedgekit/select/dist/index.cjs
var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, {enumerable: true, configurable: true, writable: true, value}) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __markAsModule = (target) => __defProp(target, "__esModule", {value: true});
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, {get: all[name], enumerable: true});
};
var __reExport = (target, module2, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== "default")
        __defProp(target, key, {get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable});
  }
  return target;
};
var __toModule = (module2) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", module2 && module2.__esModule && "default" in module2 ? {get: () => module2.default, enumerable: true} : {value: module2, enumerable: true})), module2);
};

// src/index.ts
__markAsModule(exports);
__export(exports, {
  Multiselect: () => Multiselect_default,
  Select: () => Select_default,
  StyledActionBar: () => ActionBar_default,
  StyledBadge: () => Badge_default,
  StyledDropdown: () => DropdownWrapper_default,
  StyledDropdownText: () => DropdownText_default,
  StyledFilter: () => Filter_default,
  StyledLabel: () => Label_default,
  StyledOptionList: () => OptionList_default
});

// src/components/Multiselect/Multiselect.tsx
var import_react5 = __toModule(require("react"));
var import_core = __toModule(require("@dmsi/core"));

// src/components/shared/Dropdown/Dropdown.tsx
var import_react3 = __toModule(require("react"));
var import_react_merge_refs = __toModule(require("react-merge-refs"));
var import_uuid = __toModule(require("uuid"));
var import_icons = __toModule(require("@wedgekit/icons"));
var import_info_button = __toModule(require("@wedgekit/info-button"));
var import_primitives11 = __toModule(require("@wedgekit/primitives"));
var import_utils2 = __toModule(require("@wedgekit/utils"));

// src/components/shared/Dropdown/styled/Badge.tsx
var import_styled_components = __toModule(require("styled-components"));
var import_color = __toModule(require("@wedgekit/color"));
var import_primitives = __toModule(require("@wedgekit/primitives"));
var Badge = import_styled_components.default.div`
  grid-column: 2/3;
  margin-right: ${(0, import_primitives.getBasePx)(3)};
  height: ${(0, import_primitives.getBasePx)(6)};
  border-radius: ${import_primitives.default.base * 3}px;
  text-align: center;
  background-color: ${import_color.default.B200};
  color: ${import_color.default.B600};
  font-family: ${import_primitives.default.fontFamily};
  font-weight: ${import_primitives.default.fontWeight.link};
  line-height: 1.5rem;
  font-size: ${import_primitives.default.fontSize.input}px;
  ${({children}) => {
  const charCount = (children == null ? void 0 : children.toString().length) || 0;
  return `width: ${import_primitives.default.base * (4 + 2 * charCount)}px;`;
}};
`;
var Badge_default = Badge;

// src/components/shared/Dropdown/styled/Caret.tsx
var import_styled_components2 = __toModule(require("styled-components"));
var import_primitives2 = __toModule(require("@wedgekit/primitives"));
var Caret = import_styled_components2.default.div`
  grid-column: 3/4;
  height: ${(0, import_primitives2.getBasePx)(4)};
  width: ${(0, import_primitives2.getBasePx)(4)};
`;
var Caret_default = Caret;

// src/components/shared/Dropdown/styled/DropdownWrapper.tsx
var import_styled_components3 = __toModule(require("styled-components"));
var import_color2 = __toModule(require("@wedgekit/color"));
var import_primitives3 = __toModule(require("@wedgekit/primitives"));
var Dropdown = import_styled_components3.default.div`
  cursor: pointer;
  ${import_primitives3.UpdatedInputMixin}
  align-items: center;
  display: inline-grid;
  grid-template-columns: minmax(0, 1fr) min-content min-content;
  padding: 0 ${(0, import_primitives3.getBasePx)(3)};
  width: 100%;
  border: ${({disabled, popoverOpen}) => !disabled && popoverOpen && `${import_primitives3.default.border.width.standard}px solid ${import_color2.default.B500}`};
  box-shadow: ${({disabled, popoverOpen}) => !disabled && popoverOpen && `inset 0 0 0 .5px ${import_color2.default.B500}`};
  &:focus {
  }
`;
var DropdownWrapper_default = Dropdown;

// src/components/shared/Dropdown/styled/DropdownText.ts
var import_styled_components4 = __toModule(require("styled-components"));
var import_primitives4 = __toModule(require("@wedgekit/primitives"));
var import_color3 = __toModule(require("@wedgekit/color"));
var DropdownText = (0, import_styled_components4.default)(import_primitives4.Text)`
  color: ${({disabled, isPlaceholder}) => (disabled || isPlaceholder) && import_color3.default.N400};
  grid-column: 1/2;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;
var DropdownText_default = DropdownText;

// src/components/shared/Dropdown/styled/Label.tsx
var import_styled_components5 = __toModule(require("styled-components"));
var import_primitives5 = __toModule(require("@wedgekit/primitives"));
var Label = (0, import_styled_components5.default)(import_primitives5.Label)``;
var Label_default = Label;

// src/components/shared/List/List.tsx
var import_react2 = __toModule(require("react"));
var import_button = __toModule(require("@wedgekit/button"));
var import_hooks = __toModule(require("@wedgekit/hooks"));

// src/components/shared/List/utils/getNewValue.ts
var getNewValue = ({
  multiple = false,
  options,
  filteredOptions = options,
  currentIdx,
  previousIdx,
  shiftDown,
  value
}) => {
  if (!multiple) {
    return [filteredOptions[currentIdx].id];
  }
  const [startChangeIdx, endChangeIdx] = [currentIdx, previousIdx].sort();
  return options.reduce((newValue, option) => {
    const currentlySelected = value.includes(option.id);
    const optionIdx = filteredOptions.findIndex((filteredOption) => filteredOption.id === option.id);
    if (typeof optionIdx !== "number") {
      if (currentlySelected) {
        return newValue.concat(option.id);
      }
      return newValue;
    }
    if (shiftDown) {
      const addOption2 = startChangeIdx <= optionIdx && optionIdx <= endChangeIdx;
      if (currentlySelected || addOption2) {
        return newValue.concat(option.id);
      }
    }
    const toggleOption = currentIdx === optionIdx;
    const addOption = currentlySelected || toggleOption;
    const removeOption = currentlySelected && toggleOption;
    if (addOption && !removeOption) {
      return newValue.concat(option.id);
    }
    return newValue;
  }, []);
};
var getNewValue_default = getNewValue;

// src/components/shared/List/utils/getNewIndex.ts
var getNewIndex_default = (startIdx, move, listLength) => {
  const a = startIdx + move;
  const n = listLength;
  return (a % n + n) % n;
};

// src/components/shared/List/utils/withEscapeWatcher.tsx
var import_react = __toModule(require("react"));
var withEscapeWatcher = (Component) => (props) => {
  (0, import_react.useEffect)(() => {
    const handleKeyDown = (e) => {
      if (e.key === "Escape") {
        e.stopPropagation();
        if (props.onEscape) {
          props.onEscape(e);
        }
        if (props.toggleRef && props.toggleRef.current) {
          props.toggleRef.current.focus();
        }
      }
    };
    window.addEventListener("keydown", handleKeyDown, true);
    return () => window.removeEventListener("keydown", handleKeyDown, true);
  }, []);
  return /* @__PURE__ */ import_react.default.createElement(Component, __spreadValues({}, props));
};
var withEscapeWatcher_default = withEscapeWatcher;

// src/components/shared/List/styled/ActionBar.tsx
var import_styled_components6 = __toModule(require("styled-components"));
var import_primitives6 = __toModule(require("@wedgekit/primitives"));
var ActionBar = import_styled_components6.default.div`
  position: relative;
  display: flex;
  justify-content: center;
  button {
    margin: ${(0, import_primitives6.getBasePx)(2)};
  }
  box-shadow: 0 -2px 4px 0 rgba(0, 0, 0, 0.1);
`;
var ActionBar_default = ActionBar;

// src/components/shared/List/styled/Filter.tsx
var import_styled_components7 = __toModule(require("styled-components"));
var import_input = __toModule(require("@wedgekit/input"));
var import_primitives7 = __toModule(require("@wedgekit/primitives"));
var Filter = (0, import_styled_components7.default)(import_input.default)`
  margin-top: 0px;
  margin-bottom: ${(0, import_primitives7.getBasePx)(3)};
`;
var Filter_default = Filter;

// src/components/shared/List/styled/MultiselectOption.tsx
var import_styled_components8 = __toModule(require("styled-components"));
var import_checkbox = __toModule(require("@wedgekit/checkbox"));
var import_color4 = __toModule(require("@wedgekit/color"));
var getBackgroundColor = (active, value) => {
  if (value) {
    if (active) {
      return import_color4.default.B100;
    }
    return import_color4.default.B200;
  }
  if (active) {
    return import_color4.default.N100;
  }
  return import_color4.default.N050;
};
var MultiselectOption = (0, import_styled_components8.default)(import_checkbox.default)`
  box-sizing: border-box;
  margin-bottom: 4px;
  padding: 0 4px;
  width: 100%;

  &::first-child {
    margin-top: 12px;
  }

  &:last-child {
    margin-bottom: 0px;
  }

  ${import_checkbox.StyledLabel} {
    background: ${({active, value}) => getBackgroundColor(active, value)};
    font-weight: 400;
    user-select: none;
    margin: 0 -16px;
    border-radius: 0;
    padding: 6px 12px;

    &:hover {
      background: ${({value}) => value ? import_color4.default.B100 : import_color4.default.N100};
      label {
        color: ${import_color4.default.N600};
      }
    }
  }
`;
var MultiselectOption_default = MultiselectOption;

// src/components/shared/List/styled/OptionList.tsx
var import_styled_components9 = __toModule(require("styled-components"));
var OptionList_default = import_styled_components9.default.div`
  &:focus {
    outline: none;
  }
`;

// src/components/shared/List/styled/OptionText.tsx
var import_styled_components10 = __toModule(require("styled-components"));
var import_primitives8 = __toModule(require("@wedgekit/primitives"));
var OptionText = (0, import_styled_components10.default)(import_primitives8.Text)`
  pointer-events: none;
  margin: 8px 0px;
  line-height: 1.5rem;

  &:focus {
    outline: none;
  }
`;
var OptionText_default = OptionText;

// src/components/shared/List/styled/OptionsWrapper.tsx
var import_styled_components11 = __toModule(require("styled-components"));
var import_input2 = __toModule(require("@wedgekit/input"));
var import_primitives9 = __toModule(require("@wedgekit/primitives"));
var OptionWrapper = import_styled_components11.default.div`
  overflow-y: auto;
  max-height: 30vh;
  width: ${({toggleRef}) => toggleRef.current && `${toggleRef.current.clientWidth - import_primitives9.default.base * 6}px`};
  padding: ${(0, import_primitives9.getBasePx)(3)};

  ${import_input2.InputWrapper} {
    margin: 0px;
  }
`;
var OptionsWrapper_default = OptionWrapper;

// src/components/shared/List/styled/SelectOption.tsx
var import_styled_components12 = __toModule(require("styled-components"));
var import_color5 = __toModule(require("@wedgekit/color"));
var import_primitives10 = __toModule(require("@wedgekit/primitives"));
var getBackgroundColor2 = (value, active) => {
  if (value) {
    if (active) {
      return import_color5.default.B100;
    }
    return import_color5.default.B200;
  }
  if (active) {
    return import_color5.default.N100;
  }
  return import_color5.default.N050;
};
var SelectOption = import_styled_components12.default.div`
  align-content: center;
  background: ${({value, active}) => getBackgroundColor2(value, active)};
  cursor: pointer;
  display: flex;
  margin: 0 -${(0, import_primitives10.getBasePx)(3)} ${(0, import_primitives10.getBasePx)(1)} -${(0, import_primitives10.getBasePx)(3)};
  padding: 0 ${(0, import_primitives10.getBasePx)(3)};
  ${import_primitives10.CursorMixin};
  transition: background 0.2s ease-out;

  & div {
    flex: 1;
  }

  &:focus {
    outline: none;
  }

  label {
    font-weight: normal;
    &:focus {
      color: ${import_color5.default.N600};
    }
  }

  &:hover {
    background: ${({value}) => value ? import_color5.default.B100 : import_color5.default.N100};
    label {
      color: ${import_color5.default.N600};
    }
  }
`;
var SelectOption_default = SelectOption;

// src/components/shared/List/styled/Space.tsx
var import_styled_components13 = __toModule(require("styled-components"));
var Space = import_styled_components13.default.div`
  flex: 1;
`;
var Space_default = Space;

// src/components/shared/List/styled/StyledPopover.tsx
var import_styled_components14 = __toModule(require("styled-components"));
var import_color6 = __toModule(require("@wedgekit/color"));
var import_popover = __toModule(require("@wedgekit/popover"));
var StyledPopover_default = (0, import_styled_components14.default)(import_popover.default)`
  background: ${import_color6.default.N050};
`;

// src/components/shared/List/List.tsx
var List = (props) => {
  const {
    actionBar,
    filter: displayFilter,
    filterPlaceholder,
    label,
    multiple,
    options,
    toggleRef,
    value,
    onBlur,
    onChange,
    onEscape,
    onSelect
  } = props;
  const activeOptionRef = (0, import_react2.useRef)(null);
  const internalFocusRef = (0, import_react2.useRef)(null);
  const checkboxListRef = (0, import_react2.useRef)(null);
  const filterRef = (0, import_react2.useRef)(null);
  const [filter, setFilter] = (0, import_react2.useState)("");
  const [filteredOptions, setFilteredOptions] = (0, import_react2.useState)(options.filter((option) => option.label.toLowerCase().includes(filter.toLowerCase())));
  const [activeOptionIdx, setActiveOptionIdx] = (0, import_react2.useState)(!!value === !![] ? 0 : filteredOptions.findIndex((option) => value.includes(option.id)));
  const [allSelected, setAllSelected] = (0, import_react2.useState)(value === filteredOptions.map((option) => option.id));
  (0, import_react2.useEffect)(() => {
    var _a;
    (_a = activeOptionRef.current) == null ? void 0 : _a.scrollIntoView({behavior: "smooth", block: "nearest"});
  }, [activeOptionIdx]);
  (0, import_react2.useEffect)(() => {
    if (displayFilter) {
      filterRef.current.focus();
    } else {
      checkboxListRef.current.focus();
    }
  }, []);
  (0, import_hooks.useContainerEvent)(internalFocusRef, {
    scroll: (e, inside) => {
      if (!inside && onEscape) {
        onEscape(e);
      }
    }
  });
  const handleFilterChange = (newFilter) => {
    setFilter(newFilter);
    setFilteredOptions(options.filter((option) => option.label.toLowerCase().includes(newFilter.toLowerCase())));
  };
  const handleClick = (event) => {
    const option = event.target;
    const currentIdx = parseInt(option.getAttribute("data-idx"), 10);
    const newValue = getNewValue_default({
      multiple,
      options,
      filteredOptions,
      currentIdx,
      previousIdx: activeOptionIdx,
      shiftDown: event.shiftKey,
      value
    });
    onChange(newValue, event);
    if (onSelect) {
      onSelect(event);
    }
    setActiveOptionIdx(currentIdx);
  };
  const handleKeyDown = (event) => {
    switch (event.key) {
      case "ArrowDown":
        event.preventDefault();
        setActiveOptionIdx(getNewIndex_default(activeOptionIdx, 1, filteredOptions.length));
        break;
      case "ArrowUp":
        event.preventDefault();
        setActiveOptionIdx(getNewIndex_default(activeOptionIdx, -1, filteredOptions.length));
        break;
      case "End":
        event.preventDefault();
        setActiveOptionIdx(filteredOptions.length - 1);
        break;
      case "Home":
        event.preventDefault();
        setActiveOptionIdx(0);
        break;
      case " ":
      case "SpaceBar":
        event.preventDefault();
        activeOptionRef.current.click();
        break;
      default:
    }
  };
  const handleClearAll = (event) => {
    setAllSelected(false);
    onChange([], event);
  };
  const handleSelectAll = (event) => {
    if (allSelected) {
      handleClearAll(event);
    } else {
      const allValues = filteredOptions.reduce((acc, cur) => {
        acc.push(cur.id);
        return acc;
      }, []);
      setAllSelected(true);
      onChange(allValues, event);
    }
  };
  const handleBlur = (event) => {
    const internalFocus = !!internalFocusRef.current && event.relatedTarget instanceof Node && internalFocusRef.current.contains(event.relatedTarget);
    if (!internalFocus && onBlur) {
      onBlur(event);
    }
  };
  return /* @__PURE__ */ import_react2.default.createElement(StyledPopover_default, {
    node: toggleRef.current,
    ref: internalFocusRef,
    onExit: onEscape,
    enabled: false
  }, /* @__PURE__ */ import_react2.default.createElement(OptionsWrapper_default, {
    "data-parent": label,
    toggleRef,
    tabIndex: -1,
    onBlur: handleBlur
  }, displayFilter && /* @__PURE__ */ import_react2.default.createElement(Filter_default, {
    "data-parent": label,
    fullWidth: true,
    label: `Filter ${label} select`,
    labelHidden: true,
    placeholder: filterPlaceholder || "Search",
    ref: filterRef,
    value: filter,
    onChange: (newFilter) => {
      handleFilterChange(newFilter);
    },
    onKeyDown: handleKeyDown,
    status: "default"
  }), /* @__PURE__ */ import_react2.default.createElement(OptionList_default, {
    "aria-label": label,
    "data-parent": label,
    id: `${label}-listbox`,
    role: "listbox",
    ref: checkboxListRef,
    tabIndex: displayFilter ? -1 : 0,
    onKeyDown: handleKeyDown
  }, filteredOptions.map((option, idx) => multiple ? /* @__PURE__ */ import_react2.default.createElement(MultiselectOption_default, {
    active: activeOptionIdx === idx,
    "data-idx": idx,
    "data-parent": label,
    disabled: option.disabled,
    id: `${label}-${option.id}`,
    key: `option-${option.id}`,
    label: option.label,
    name: option.id,
    ref: idx === activeOptionIdx ? activeOptionRef : void 0,
    role: "option",
    tabIndex: -1,
    value: multiple ? value.includes(option.id) : value[0] === option.id,
    onChange: () => {
    },
    onClick: handleClick
  }) : /* @__PURE__ */ import_react2.default.createElement(SelectOption_default, {
    active: activeOptionIdx === idx,
    "aria-label": option.label,
    "aria-selected": value[0] === option.id,
    "data-parent": label,
    "data-idx": idx,
    disabled: option.disabled,
    id: `${option.id}`,
    key: `option-${option.id}`,
    ref: idx === activeOptionIdx ? activeOptionRef : void 0,
    role: "option",
    tabIndex: -1,
    value: value[0] === option.id,
    onClick: handleClick
  }, /* @__PURE__ */ import_react2.default.createElement(OptionText_default, {
    "data-parent": label,
    tabIndex: -1
  }, option.label))))), multiple && !!actionBar && /* @__PURE__ */ import_react2.default.createElement(ActionBar_default, {
    "data-parent": label,
    toggleRef,
    onBlur: handleBlur
  }, (typeof actionBar === "boolean" || actionBar.selectAll) && /* @__PURE__ */ import_react2.default.createElement(import_button.default, {
    "data-parent": label,
    domain: "secondary",
    variant: "noFill",
    onClick: handleSelectAll
  }, typeof actionBar === "object" && actionBar.selectAll && typeof actionBar.selectAll === "string" ? actionBar.selectAll : "Select All"), /* @__PURE__ */ import_react2.default.createElement(Space_default, null), (typeof actionBar === "boolean" || actionBar.clearAll) && /* @__PURE__ */ import_react2.default.createElement(import_button.default, {
    "data-parent": label,
    domain: "primary",
    variant: "noFill",
    onClick: handleClearAll
  }, typeof actionBar === "object" && actionBar.clearAll && typeof actionBar.clearAll === "string" ? actionBar.clearAll : "Clear All")));
};
var List_default = withEscapeWatcher_default(List);

// src/components/shared/utils/getMultiselectLabel.ts
var getMultiselectLabel_default = ({options, value, placeholder}) => {
  if (!value || value.length === 0) {
    return placeholder;
  }
  return value.reduce((acc, cur) => {
    const valueIsOption = options.find((option) => option.id === cur);
    if (!valueIsOption) {
      return acc === "" ? placeholder : acc;
    }
    const name = valueIsOption.label;
    if (acc !== "") {
      return `${acc}, ${name}`;
    }
    return `${name}`;
  }, ``);
};

// src/components/shared/utils/getSelectLabel.ts
var getSelectLabel_default = ({options, value, placeholder}) => {
  const valueIsOption = options.find((option) => option.id === value[0]);
  if (typeof value === "undefined" || !valueIsOption) {
    return placeholder;
  }
  return valueIsOption.label;
};

// src/components/shared/Dropdown/Dropdown.tsx
var Dropdown2 = import_react3.default.forwardRef((props, forwardedRef) => {
  const _a = props, {className} = _a, otherProps = __objRest(_a, ["className"]);
  if (props.invalid) {
    console.warn(`prop 'invalid' has been deprecated in favor of passing 'status'="error"`);
    console.warn(`To be removed in @wedgekit/select@8.0.0`);
  }
  const internalRef = (0, import_react3.useRef)(null);
  const focusRef = (0, import_react3.useRef)(null);
  const dropdownRefs = (0, import_react_merge_refs.default)([forwardedRef, internalRef]);
  const uuidRef = (0, import_react3.useRef)((0, import_uuid.v4)());
  const [showPopover, setShowPopover] = (0, import_react3.useState)(false);
  const getValueCount = () => {
    if (typeof props.value === "undefined") {
      return 0;
    }
    const validValues = props.options.map((option) => option.id);
    const values = Array.isArray(props.value) ? props.value : [props.value];
    return values.filter((val) => validValues.includes(val)).length;
  };
  const dropdownProps = {};
  if (props.labelHidden) {
    dropdownProps["aria-label"] = props.label;
  }
  const isInternal = (event) => !!focusRef.current && event.relatedTarget instanceof Node && focusRef.current.contains(event.relatedTarget);
  const handleDropdownFocus = (event) => {
    if (props.onFocus && !isInternal(event)) {
      props.onFocus(event);
    }
  };
  const handleDropdownBlur = (event) => {
    if (props.onBlur && event.currentTarget === event.target && !isInternal(event)) {
      props.onBlur(event);
    }
  };
  const handleListBlur = (event) => {
    if (props.onBlur) {
      props.onBlur(event);
    }
    if (event.relatedTarget !== internalRef.current) {
      setShowPopover(false);
    }
  };
  return /* @__PURE__ */ import_react3.default.createElement(import_primitives11.FieldStatusWrapper, {
    className,
    disabled: props.disabled,
    fullLength: true,
    readOnly: props.readOnly
  }, !props.labelHidden && /* @__PURE__ */ import_react3.default.createElement(import_primitives11.Row, {
    center: true
  }, /* @__PURE__ */ import_react3.default.createElement(Label_default, {
    disabled: props.disabled,
    htmlFor: uuidRef.current,
    required: props.required
  }, props.label), props.info && /* @__PURE__ */ import_react3.default.createElement(import_info_button.default, {
    info: props.info,
    tabIndex: props.tabIndex
  })), /* @__PURE__ */ import_react3.default.createElement("div", {
    ref: focusRef
  }, /* @__PURE__ */ import_react3.default.createElement(DropdownWrapper_default, __spreadProps(__spreadValues({}, dropdownProps), {
    labelHidden: props.labelHidden,
    disabled: props.disabled,
    invalid: props.invalid,
    popoverOpen: showPopover,
    readOnly: props.readOnly,
    ref: dropdownRefs,
    role: "button",
    status: props.status,
    tabIndex: props.disabled || showPopover ? import_primitives11.default.tabIndex.disabled : props.tabIndex || 0,
    onBlur: handleDropdownBlur,
    onFocus: handleDropdownFocus,
    onClick: () => setShowPopover(!showPopover),
    onKeyPress: (e) => {
      if (!showPopover) {
        return (0, import_utils2.handleKeyEvent)({
          "Enter|\\s": () => {
            setShowPopover(true);
          }
        })(e);
      }
      return null;
    }
  }), /* @__PURE__ */ import_react3.default.createElement(DropdownText_default, {
    htmlFor: uuidRef.current,
    disabled: props.disabled,
    required: props.required,
    isPlaceholder: !props.value || !props.value.length
  }, props.multiple ? getMultiselectLabel_default({
    options: props.options,
    placeholder: props.placeholder || "Select option(s)",
    value: props.value
  }) : getSelectLabel_default({
    options: props.options,
    placeholder: props.placeholder || "Select one",
    value: props.value
  })), props.multiple && !!props.value.length && getValueCount() > 0 && /* @__PURE__ */ import_react3.default.createElement(Badge_default, null, getValueCount()), /* @__PURE__ */ import_react3.default.createElement(Caret_default, null, /* @__PURE__ */ import_react3.default.createElement(import_icons.Chevron, {
    color: props.disabled ? "N400" : "N600",
    direction: showPopover && !props.disabled ? "up" : "down"
  }))), showPopover && !props.disabled && /* @__PURE__ */ import_react3.default.createElement(List_default, __spreadProps(__spreadValues({}, otherProps), {
    id: uuidRef.current,
    toggleRef: internalRef,
    onBlur: handleListBlur,
    onEscape: () => {
      setShowPopover(false);
    },
    onSelect: props.multiple ? () => {
    } : (e) => {
      var _a2;
      (_a2 = internalRef.current) == null ? void 0 : _a2.focus();
      setShowPopover(false);
      if (props.onFocus) {
        props.onFocus(e);
      }
    },
    "data-cy": "multiselect-list"
  }))));
});
var Dropdown_default = Dropdown2;

// src/components/shared/Native/Native.tsx
var import_react4 = __toModule(require("react"));
var import_uuid2 = __toModule(require("uuid"));
var import_icons2 = __toModule(require("@wedgekit/icons"));
var import_info_button2 = __toModule(require("@wedgekit/info-button"));
var import_primitives13 = __toModule(require("@wedgekit/primitives"));

// src/components/shared/Native/styled/Wrapper.tsx
var import_styled_components15 = __toModule(require("styled-components"));
var MobileSelectWrapper = import_styled_components15.default.div`
  position: relative;
  z-index: 0;
`;
var Wrapper_default = MobileSelectWrapper;

// src/components/shared/Native/styled/NativeDropdown.tsx
var import_styled_components16 = __toModule(require("styled-components"));
var NativeDropdown_default = (0, import_styled_components16.default)(DropdownWrapper_default)`
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
`;

// src/components/shared/Native/styled/SelectRaw.tsx
var import_styled_components17 = __toModule(require("styled-components"));
var import_primitives12 = __toModule(require("@wedgekit/primitives"));
var SelectRaw_default = import_styled_components17.default.select`
  ${import_primitives12.UpdatedInputMixin};
  appearance: none;
  color: transparent;
  background-color: transparent;
  box-shadow: none;
  width: 100%;
`;

// src/components/shared/Native/Native.tsx
var NativeSelect = import_react4.default.forwardRef((props, forwardedRef) => {
  const uuidRef = (0, import_react4.useRef)((0, import_uuid2.v4)());
  const [chevronDown, setChevronDown] = (0, import_react4.useState)(true);
  const dropdownProps = {};
  if (props.labelHidden) {
    dropdownProps["aria-label"] = props.label;
  }
  return /* @__PURE__ */ import_react4.default.createElement(import_primitives13.FieldStatusWrapper, {
    className: props.className,
    disabled: props.disabled,
    readOnly: props.readOnly
  }, !props.labelHidden && /* @__PURE__ */ import_react4.default.createElement(import_primitives13.Row, {
    center: true
  }, /* @__PURE__ */ import_react4.default.createElement(Label_default, {
    disabled: props.disabled,
    htmlFor: uuidRef.current,
    required: props.required
  }, props.label), props.info && /* @__PURE__ */ import_react4.default.createElement(import_info_button2.default, {
    info: props.info,
    tabIndex: props.tabIndex
  })), /* @__PURE__ */ import_react4.default.createElement(Wrapper_default, null, /* @__PURE__ */ import_react4.default.createElement(SelectRaw_default, __spreadValues({
    multiple: true,
    required: props.required,
    onBlur: (e) => {
      setChevronDown(true);
      if (props.onBlur) {
        props.onBlur(e);
      }
    },
    onChange: (e) => props.onChange([...e.target.selectedOptions].map((option) => option.value), e),
    onFocus: (e) => {
      setChevronDown(false);
      if (props.onFocus) {
        props.onFocus(e);
      }
    }
  }, dropdownProps), props.options.map((option) => /* @__PURE__ */ import_react4.default.createElement("option", {
    value: option.id,
    selected: !!props.value.find((value) => value === option.id),
    disabled: option.disabled
  }, option.label))), /* @__PURE__ */ import_react4.default.createElement(NativeDropdown_default, {
    disabled: props.disabled,
    invalid: props.invalid,
    labelHidden: props.labelHidden,
    readOnly: props.readOnly,
    ref: forwardedRef,
    status: props.status
  }, /* @__PURE__ */ import_react4.default.createElement(DropdownText_default, {
    htmlFor: uuidRef.current,
    disabled: props.disabled,
    required: props.required
  }, props.multiple ? getMultiselectLabel_default({
    options: props.options,
    placeholder: props.placeholder || "Select option(s)",
    value: props.value
  }) : getSelectLabel_default({
    options: props.options,
    placeholder: props.placeholder || "Select one",
    value: props.value
  })), /* @__PURE__ */ import_react4.default.createElement(Caret_default, null, /* @__PURE__ */ import_react4.default.createElement(import_icons2.Chevron, {
    color: props.disabled ? "N400" : "N600",
    direction: chevronDown ? "down" : "up"
  })))));
});
var Native_default = NativeSelect;

// src/components/Multiselect/Multiselect.tsx
var Multiselect = import_react5.default.forwardRef((props, ref) => /* @__PURE__ */ import_react5.default.createElement(import_react5.default.Fragment, null, import_core.browser.mobile ? /* @__PURE__ */ import_react5.default.createElement(Native_default, __spreadProps(__spreadValues({}, props), {
  multiple: true,
  ref
})) : /* @__PURE__ */ import_react5.default.createElement(Dropdown_default, __spreadProps(__spreadValues({}, props), {
  multiple: true,
  ref
}))));
var Multiselect_default = Multiselect;

// src/components/Select/Select.tsx
var import_react6 = __toModule(require("react"));
var import_core2 = __toModule(require("@dmsi/core"));
var Select = import_react6.default.forwardRef((props, ref) => {
  const options = props.delible ? [
    {
      id: typeof props.delible === "object" ? props.delible.id : "DELIBLE_OPTION_ID",
      label: typeof props.delible === "object" && props.delible.label || "---"
    },
    ...props.options
  ] : props.options;
  const value = [props.value];
  const onChange = (newValue, event) => {
    props.onChange(newValue[0] || "", event);
  };
  return /* @__PURE__ */ import_react6.default.createElement(import_react6.default.Fragment, null, import_core2.browser.mobile ? /* @__PURE__ */ import_react6.default.createElement(Native_default, __spreadProps(__spreadValues({}, props), {
    options,
    ref,
    value,
    onChange
  })) : /* @__PURE__ */ import_react6.default.createElement(Dropdown_default, __spreadProps(__spreadValues({}, props), {
    options,
    ref,
    value,
    onChange
  })));
});
var Select_default = Select;
