// @wedgekit/toast/dist/index.cjs
var M=Object.create,m=Object.defineProperty;var O=Object.getOwnPropertyDescriptor;var Q=Object.getOwnPropertyNames;var U=Object.getPrototypeOf,V=Object.prototype.hasOwnProperty;var v=e=>m(e,"__esModule",{value:!0});var X=(e,r)=>{for(var a in r)m(e,a,{get:r[a],enumerable:!0})},Y=(e,r,a)=>{if(r&&typeof r=="object"||typeof r=="function")for(let p of Q(r))!V.call(e,p)&&p!=="default"&&m(e,p,{get:()=>r[p],enumerable:!(a=O(r,p))||a.enumerable});return e},t=e=>Y(v(m(e!=null?M(U(e)):{},"default",e&&e.__esModule&&"default"in e?{get:()=>e.default,enumerable:!0}:{value:e,enumerable:!0})),e);v(exports);X(exports,{default:()=>J});var o=t(require("react")),l=t(require("@wedgekit/icons")),g=t(require("@wedgekit/layout"));var B=t(require("styled-components")),S=t(require("@wedgekit/layout")),L=t(require("@wedgekit/color")),Z=(0,B.default)(S.default.Section)`
  align-self: stretch;
  background-color: ${({crust:e})=>L.default[e!=null?e:"B700"]};
  border-radius: 100px;
  width: 5px;
`,x=Z;var N=t(require("styled-components")),y=t(require("@wedgekit/color")),C=t(require("@wedgekit/icons")),_=(0,N.default)(C.InteractiveWrapper)`
  align-content: center;
  box-sizing: border-box;
  display: grid;
  height: 42px;
  justify-content: center;
  margin-right: 18px;
  width: 42px;

  &:hover,
  &:focus {
    background-color: ${y.default.N100};
    svg {
      fill: ${y.default.N600};
    }
  }
`,u=_;var j=t(require("styled-components")),n=t(require("@wedgekit/color")),z=t(require("@wedgekit/layout")),k=t(require("@wedgekit/primitives")),R=(0,j.default)(z.default.Section)`
  align-content: center;
  height: ${(0,k.getBasePx)(8)};
  width: ${(0,k.getBasePx)(8)};
  background-color: ${({bg:e})=>e.startsWith("R")?(0,n.hexToRGBA)(n.default[e],.8):n.default[e]};
  border-radius: 50%;
  display: grid;
  margin-left: ${e=>e?18:16}px;
  justify-content: center;
`,w=R;var G=t(require("styled-components")),c=t(require("@wedgekit/primitives")),ee=(0,G.default)(c.Text)`
  line-height: 1.5em;
  margin: ${(0,c.getBasePx)(2)} ${(0,c.getBasePx)(6)};
`,h=ee;var A=t(require("styled-components")),q=t(require("@wedgekit/color")),s=t(require("@wedgekit/primitives")),te=(0,A.default)(s.Title)`
  color: ${q.default.N600};
  margin: ${(0,s.getBasePx)(2)} ${(0,s.getBasePx)(6)};
`,f=te;var D=t(require("styled-components")),T=D.default.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;var E=t(require("styled-components")),F=t(require("@wedgekit/layout")),H=t(require("@wedgekit/color")),d=t(require("@wedgekit/primitives")),oe=(0,E.default)(F.default.Grid)`
  background-color: ${H.default.N050};
  border-radius: 2px;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
  display: grid;
  grid-column-gap: 0;
  grid-row-gap: 0;
  position: fixed;
  bottom: ${(0,d.getBasePx)(4)};
  padding: ${(0,d.getBasePx)(2)};
  width: ${d.default.widths.toast}px;
  z-index: 1;
`,W=oe;var re=({className:e,header:r,crust:a,icon:p,message:b,remove:K,domainColors:i,extraHeaderContent:P})=>{var $,I;return o.default.createElement(W,{className:e,areas:["bar icon header close","bar icon message close"],columns:["max-content","max-content",1,"max-content"],align:"center"},a&&o.default.createElement(x,{area:"bar",crust:i==null?void 0:i.crust}),p&&o.default.createElement(w,{bg:($=i==null?void 0:i.bg)!=null?$:"B100",area:"icon"},o.default.createElement(l.IconWidth,{iconWidth:16},o.default.createElement(l.default,{color:(I=i==null?void 0:i.icon)!=null?I:"B700"},p))),o.default.createElement(g.default.Section,{area:"header"},P?o.default.createElement(T,null,o.default.createElement(f,{level:6},r),P):o.default.createElement(f,{level:6},r)),b&&o.default.createElement(g.default.Section,{area:"message"},o.default.createElement(h,null,b)),o.default.createElement(g.default.Section,{area:"close"},o.default.createElement(u,{onClick:K},o.default.createElement(l.IconWidth,{iconWidth:16},o.default.createElement(l.default,null,"close")))))},J=re;
