// @wedgekit/info-button/dist/index.cjs
var O=Object.create,l=Object.defineProperty;var j=Object.getOwnPropertyDescriptor;var W=Object.getOwnPropertyNames;var _=Object.getPrototypeOf,z=Object.prototype.hasOwnProperty;var w=o=>l(o,"__esModule",{value:!0});var H=(o,r)=>{for(var p in r)l(o,p,{get:r[p],enumerable:!0})},M=(o,r,p)=>{if(r&&typeof r=="object"||typeof r=="function")for(let n of W(r))!z.call(o,n)&&n!=="default"&&l(o,n,{get:()=>r[n],enumerable:!(p=j(r,n))||p.enumerable});return o},t=o=>M(w(l(o!=null?O(_(o)):{},"default",o&&o.__esModule&&"default"in o?{get:()=>o.default,enumerable:!0}:{value:o,enumerable:!0})),o);w(exports);H(exports,{default:()=>L});var e=t(require("react")),d=t(require("@wedgekit/primitives"));var a=t(require("styled-components")),T=t(require("@wedgekit/button")),i=t(require("@wedgekit/color")),c=t(require("@wedgekit/primitives")),f=a.default.div`
  margin: -1px 0;
  padding: 1px;
  border: solid 1px;
  border-radius: 50%;
  border-color: transparent;
  width: max-content;

  &:focus-within {
    border-color: ${i.default.B500};
  }
`,u=(0,a.default)(T.IconButton)`
  background: ${({background:o})=>o&&i.default[o]||i.default.N100};
  width: ${c.default.fontSize.label}px;
  height: ${c.default.fontSize.label}px;
  padding: 0;

  &:hover:not([disabled]) {
    background: ${({background:o})=>o&&i.default[o]||i.default.N100};
  }

  &:focus {
    background: ${({background:o})=>o&&i.default[o]||i.default.N100};
  }
`;var B=t(require("styled-components")),h=t(require("@wedgekit/color")),I=t(require("@wedgekit/popover")),g=t(require("@wedgekit/primitives")),D=(0,B.default)(I.default)`
  padding: ${(0,g.getBasePx)(4)};
  display: grid;
  grid-gap: ${(0,g.getBasePx)(4)};
  background: ${h.default.N700};
  border-radius: 2px;
  max-width: ${o=>`${o.width}px`};
`,k=D;var P=t(require("styled-components")),$=t(require("@wedgekit/primitives")),F=(0,P.default)($.Link)`
  justify-self: end;
`,b=F;var C=t(require("styled-components")),S=t(require("@wedgekit/color")),E=t(require("@wedgekit/primitives")),K=(0,C.default)(E.Text)`
  color: ${S.default.N050};
`,x=K;var V=300,q=({info:o,popoverClassname:r,tabIndex:p,onClick:n,onExit:v})=>{let[N,y]=(0,e.useState)(!1),s=(0,e.useRef)(null);return e.default.createElement(e.default.Fragment,null,e.default.createElement(f,null,e.default.createElement(u,{"aria-label":`${o.title} info`,iconColor:o.iconColor||"N600",background:o.background,icon:"info",label:o.title,tabIndex:p,type:"button",onClick:m=>(y(!0),n&&n(m)),ref:s})),N&&s.current&&e.default.createElement(k,{bearings:{side:"right",align:"center",offset:parseFloat((0,d.getBasePx)(2))},className:r,node:s.current,width:V,onExit:m=>(y(!1),v&&v(m))},e.default.createElement(d.Title,{level:3,white:!0},o.title),o.content&&e.default.createElement(x,null,o.content),o.link&&e.default.createElement(b,{dark:!0,href:o.link.url,target:"_blank"},o.link.label)))},L=q;
