// @wedgekit/scripts/bin/src/scripts/create/utils/createPackage.js
/* eslint-disable jira-ticket-todos/jira-ticket-todos */
import version from 'latest-version';
const generateDescription = (options) => {
    // TODO: Find out if there is a preferred default description
    if (options.desc)
        return options.desc;
    if (options.script)
        return 'DMSi node script';
    return 'DMSi Agility Application';
};
const getScripts = (options) => ({
    start: `wedgekit start${options.script ? ' -s' : ''}`,
    build: `wedgekit build${options.script ? ' -s' : ''}`,
});
const getDependencies = async (options) => {
    const applicationPackages = {
        react: '17.0.2',
        'react-dom': '17.0.2',
        'styled-components': '4.3.1',
    };
    if (options.script)
        return {
            '@wedgekit/color': await version('@wedgekit/color'),
            chalk: '^4.1.1',
        };
    if (options.template === 'basic') {
        return applicationPackages;
    }
    return {
        ...applicationPackages,
        'react-router-dom': '5.3.0',
    };
};
const getDevDependencies = (options) => {
    if (options.script)
        return {};
    return {
        '@types/react': '17.0.3',
        '@types/react-dom': '17.0.3',
        vite: '2.4.1',
    };
};
export default async (options) => ({
    name: options.name,
    version: '0.0.0',
    description: generateDescription(options),
    license: 'MIT',
    private: true,
    scripts: {
        test: 'ava',
        ...getScripts(options),
    },
    dependencies: {
        ...(await getDependencies(options)),
    },
    devDependencies: {
        '@wedgekit/scripts': 'latest',
        ava: '3.15.0',
        'ts-node': '9.1.1',
        typescript: '4.3.4',
        ...getDevDependencies(options),
    },
    type: 'module',
});
