// @wedgekit/scripts/bin/configs/rollup-app-assets-plugin.js
import { extname } from 'path';

// Generate app-assets.json for The Force to consume.  Lightly adapted from rollup-html-plugin.
export default function appAssetsPlugin() {
  return {
    name: 'app-assets',
    async generateBundle(output, bundle) {
      const files = Object.values(bundle).filter(
        (file) => file.type === 'chunk' || file.type === 'asset',
      );
      const result = { js: [], css: [] };

      files.forEach((file) => {
        const { fileName } = file;
        const extension = extname(fileName).substring(1);
        result[extension] = (result[extension] || []).concat(file);
      });

      const appAssets = {
        css: result.css.map((cssFile) => cssFile.fileName),
        js: result.js.filter((jsFile) => jsFile.isEntry).map((entryFile) => entryFile.fileName),
        esm: output.format === 'es',
      };

      const assetFile = {
        type: 'asset',
        source: JSON.stringify(appAssets, null, 2),
        name: 'The Force App Assets',
        fileName: 'app-assets.json',
      };

      this.emitFile(assetFile);
    },
  };
}
