// @wedgekit/palette/src/index.ts
const colors = {
  /**
   * 700 (darkest) level of dark blue
   */
  D700: '#002131' as const,
  /**
   * 600 level of dark blue
   */
  D600: '#00354E' as const,
  /**
   * 500 level of dark blue
   */
  D500: '#00486B' as const,
  /**
   * 400 level of dark blue
   */
  D400: '#457993' as const,
  /**
   * 300 level of dark blue
   */
  D300: '#8BABBB' as const,
  /**
   * 200 level of dark blue
   */
  D200: '#D0DDE4' as const,
  /**
   * 100 (lightest) level of dark blue
   */
  D100: '#E7EEF1' as const,
  /**
   * 700 (darkest) level of blue
   */
  B700: '#00365A' as const,
  /**
   * 600 level of blue
   */
  B600: '#00568F' as const,
  /**
   * 500 level of blue
   */
  B500: '#0075C4' as const,
  /**
   * 400 level of blue
   */
  B400: '#459AD4' as const,
  /**
   * 300 level of blue
   */
  B300: '#8BC0E4' as const,
  /**
   * 200 level of blue
   */
  B200: '#D0E5F4' as const,
  /**
   * 100 (lightest) level of blue
   */
  B100: '#E7F2F9' as const,
  /**
   * 700 (darkest) level of green
   */
  G700: '#126240' as const,
  /**
   * 600 level of green
   */
  G600: '#1D9C66' as const,
  /**
   * 500 level of green
   */
  G500: '#27D68B' as const,
  /**
   * 400 level of green
   */
  G400: '#61E1AA' as const,
  /**
   * 300 level of green
   */
  G300: '#9CECCA' as const,
  /**
   * 200 level of green
   */
  G200: '#D7F7E9' as const,
  /**
   * 100 (lightest) level of green
   */
  G100: '#EEFBF6' as const,
  /**
   * 700 (darkest) level of red
   */
  R700: '#892615' as const,
  /**
   * 600 level of red
   */
  R600: '#B0311B' as const,
  /**
   * 500 level of red
   */
  R500: '#D63B20' as const,
  /**
   * 400 level of red
   */
  R400: '#DD5E48' as const,
  /**
   * 300 level of red
   */
  R300: '#E48271' as const,
  /**
   * 200 level of red
   */
  R200: '#F0B7AD' as const,
  /**
   * 100 (lightest) level of red
   */
  R100: '#FBEDEA' as const,
  /**
   * 700 (darkest) level of yellow
   */
  Y700: '#996C0D' as const,
  /**
   * 600 level of yellow
   */
  Y600: '#C58B10' as const,
  /**
   * 500 level of yellow
   */
  Y500: '#F0A913' as const,
  /**
   * 400 level of yellow
   */
  Y400: '#F4C053' as const,
  /**
   * 300 level of yellow
   */
  Y300: '#F6D07E' as const,
  /**
   * 200 level of yellow
   */
  Y200: '#F9DFA9' as const,
  /**
   * 100 (lightest) level of yellow
   */
  Y100: '#FCEFD4' as const,
  /**
   * 700 (darkest) level of greyscale
   */
  N700: '#353636' as const,
  /**
   * 600 level of greyscale
   */
  N600: '#555656' as const,
  /**
   * 500 level of greyscale
   */
  N500: '#6B6D6D' as const,
  /**
   * 400 level of greyscale
   */
  N400: '#999B9B' as const,
  /**
   * 300 level of greyscale
   */
  N300: '#BFC0C0' as const,
  /**
   * 200 level of greyscale
   */
  N200: '#D9D9D9' as const,
  /**
   * 100 level of greyscale
   */
  N100: '#F2F2F2' as const,
  /**
   * 50 (lightest) level of greyscale
   */
  N050: '#FFFFFF' as const,
  /**
   * tag color
   */
  T100: '#005BA6' as const,
  /**
   * tag color
   */
  T110: '#00AEEF' as const,
  /**
   * tag color
   */
  T200: '#00950B' as const,
  /**
   * tag color
   */
  T210: '#72EE59' as const,
  /**
   * tag color
   */
  T300: '#FFFF00' as const,
  /**
   * tag color
   */
  T310: '#F7941D' as const,
  /**
   * tag color
   */
  T400: '#FF0000' as const,
  /**
   * tag color
   */
  T410: '#C12275' as const,
  /**
   * tag color
   */
  T420: '#FA8CBF' as const,
  /**
   * tag color
   */
  T500: '#762D91' as const,
  /**
   * tag color
   */
  T600: '#754C24' as const,
  /**
   * tag color
   */
  T700: '#000000' as const,

  /**
   * START Denim color family
   */
  /* Darkest denim */
  DE700: '#00486B' as const,
  DE650: '#336D89' as const,
  DE400: '#8DADBD' as const,
  DE100: '#D4DFE5' as const,
  DE050: '#EBF0F3' as const,
  /**
   * END Denim color family
   */

  /**
   * START Graphite color family
   */
  GR700: '#353636' as const,
  GR650: '#5D5E5E' as const,
  GR400: '#A5A5A5' as const,
  GR100: '#DCDDDD' as const,
  GR050: '#EFEFEF' as const,
  /**
   * END Graphite color family
   */

  /**
   * START Sage color family
   */
  SA700: '#006B58' as const,
  SA650: '#338979' as const,
  SA400: '#8DBDB4' as const,
  SA100: '#D4E5E3' as const,
  SA050: '#EBF3F2' as const,
  /**
   * END Sage color family
   */

  /**
   * START Terra Cotta color family
   */
  TC700: '#B44100' as const,
  TC650: '#C36733' as const,
  TC450: '#FF9458' as const,
  TC400: '#DDAA8D' as const,
  TC100: '#F2DFD4' as const,
  TC050: '#F9F0EB' as const,
  /**
   * END Terra Cotta color family
   */

  /**
   * START Violet color family
   */
  VI700: '#4E008B' as const,
  VI650: '#7233A2' as const,
  VI400: '#B899D1' as const,
  VI100: '#E0D4EB' as const,
  VI050: '#F1EBF6' as const,
  /**
   * END Violet color family
   */

  /**
   * START Azure color family
   */
  AZ700: '#003DB4' as const,
  AZ650: '#3364C3' as const,
  AZ450: '#689BFF' as const,
  AZ400: '#8DA8DD' as const,
  AZ100: '#D4DDF2' as const,
  AZ050: '#EBEFF9' as const,
  /**
   * END Azure color family
   */

  /**
   * START Carmine color family
   */
  CA700: '#960018' as const,
  CA650: '#BF0020' as const,
  CA400: '#FF7089' as const,
  CA150: '#FFC3CE' as const,
  /**
   * END Carmine color family
   */
};

export type ColorPalette = Readonly<typeof colors>;

export type ColorTag = keyof ColorPalette;

export default colors;
