// @wedgekit/button/dist/index.cjs
var oo=Object.create,b=Object.defineProperty;var to=Object.getOwnPropertyDescriptor;var eo=Object.getOwnPropertyNames,y=Object.getOwnPropertySymbols,ro=Object.getPrototypeOf,V=Object.prototype.hasOwnProperty,F=Object.prototype.propertyIsEnumerable;var H=(t,e,n)=>e in t?b(t,e,{enumerable:!0,configurable:!0,writable:!0,value:n}):t[e]=n,v=(t,e)=>{for(var n in e||(e={}))V.call(e,n)&&H(t,n,e[n]);if(y)for(var n of y(e))F.call(e,n)&&H(t,n,e[n]);return t};var L=t=>b(t,"__esModule",{value:!0});var T=(t,e)=>{var n={};for(var i in t)V.call(t,i)&&e.indexOf(i)<0&&(n[i]=t[i]);if(t!=null&&y)for(var i of y(t))e.indexOf(i)<0&&F.call(t,i)&&(n[i]=t[i]);return n};var no=(t,e)=>{for(var n in e)b(t,n,{get:e[n],enumerable:!0})},io=(t,e,n)=>{if(e&&typeof e=="object"||typeof e=="function")for(let i of eo(e))!V.call(t,i)&&i!=="default"&&b(t,i,{get:()=>e[i],enumerable:!(n=to(e,i))||n.enumerable});return t},s=t=>io(L(b(t!=null?oo(ro(t)):{},"default",t&&t.__esModule&&"default"in t?{get:()=>t.default,enumerable:!0}:{value:t,enumerable:!0})),t);L(exports);no(exports,{IconButton:()=>S,default:()=>C});var u=s(require("react")),X=s(require("@wedgekit/indicators"));var Y=s(require("styled-components")),ao=Y.default.div`
  align-items: center;
  display: grid;
  justify-items: center;
`,O=ao;var j=s(require("styled-components")),x=s(require("@wedgekit/primitives")),so=j.default.button`
  align-items: center;
  appearance: none;
  background: ${t=>t.domain.bg};
  border: ${t=>t.domain.border};
  border-radius: 4px;
  color: ${t=>t.domain.text};
  cursor: pointer;
  display: flex;
  font-size: ${x.default.fontSize.body}px;
  font-weight: ${x.default.fontWeight.link};
  height: 40px;
  justify-content: center;
  line-height: 1.5rem;
  padding: 0 14.5px;
  position: relative;
  text-transform: capitalize;
  transition: all 0.2s ease-in-out;
  width: ${t=>t.fullWidth&&"100%"};

  &:disabled {
    background: ${t=>t.domain.disabled};
    border: ${t=>t.domain.disabledBorder};
    opacity: 0.5;
  }

  &:focus {
    background: ${t=>t.domain.focus};
    border: ${t=>t.domain.focusOuterBorder};
    box-shadow: ${t=>t.domain.focusInnerBorder};
    color: ${t=>t.domain.focusText};
    outline: none;
    transition: all 0.2s ease-in-out;
  }

  &:hover:not([disabled]) {
    background: ${t=>t.domain.hover};
    border: ${t=>t.domain.hoverBorder};
    box-shadow: ${t=>t.domain.dropShadow};
    color: ${t=>t.domain.hoverText};
    transition: all 0.2s ease-in-out;
  }

  &:hover:focus {
    border: ${t=>t.domain.focusOuterBorder};
    box-shadow: ${t=>t.domain.focusInnerBorder};
    background: ${t=>t.domain.focus};
  }

  &:focus svg {
    fill: ${t=>t.domain.focusText};
  }

  &:hover:enabled svg {
    fill: ${t=>t.domain.hoverText};
  }

  ${x.CursorMixin};
`,R=so;var z=s(require("styled-components")),A=s(require("@wedgekit/icons")),lo=(0,z.default)(A.default)`
  display: flex;
  margin: ${({childrenExist:t})=>t?"0 8px 0 0":"0 -4px"};
  width: 16px;
  height: 16px;

  svg {
    fill: ${t=>t.domain.text};
  }
`,W=lo;var J=s(require("styled-components")),uo=J.default.div`
  align-items: center;
  display: flex;
  grid-column: 1;
  grid-row: 1;
  justify-content: center;
  visibility: ${({visible:t})=>t?"visible":"hidden"};
`,I=uo;var o=s(require("@wedgekit/color")),d="1.5px solid transparent",a=t=>`0px 0px ${t==="lighter"?` 2px ${o.default.T700}40`:` 4px ${o.default.T700}1F`}`,r=t=>`1.5px solid ${t}`,l=t=>`inset 0px 0px 0px 1px ${t}`,co={primary:{neutral:{bg:o.default.B500,border:d,disabled:o.default.B500,disabledBorder:d,dropShadow:a("lighter"),focus:o.default.B500,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.B600),focusText:o.default.N050,hover:o.default.B600,hoverBorder:a("lighter"),hoverText:o.default.N050,text:o.default.N050,textTag:"N050"},noFill:{bg:"transparent",border:d,disabled:"none",disabledBorder:d,dropShadow:"none",focus:"transparent",focusInnerBorder:l("transparent"),focusOuterBorder:r(o.default.B300),focusText:o.default.B500,hover:o.default.B100,hoverBorder:d,hoverText:o.default.B600,text:o.default.B500,textTag:"B500"},outline:{bg:"transparent",border:r(o.default.B500),disabled:"none",disabledBorder:r(o.default.B500),dropShadow:a("darker"),focus:o.default.B200,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.B600),focusText:o.default.B600,hover:o.default.B100,hoverBorder:r(o.default.B600),hoverText:o.default.B600,text:o.default.B500,textTag:"B500"}},secondary:{neutral:{bg:o.default.N100,border:d,disabled:o.default.N100,disabledBorder:d,dropShadow:a("lighter"),focus:o.default.N200,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.N400),focusText:o.default.N600,hover:o.default.N200,hoverBorder:a("lighter"),hoverText:o.default.N600,text:o.default.N600,textTag:"N600"},noFill:{border:d,bg:"transparent",disabled:"none",disabledBorder:d,dropShadow:"none",focus:"transparent",focusInnerBorder:l("transparent"),focusOuterBorder:r(o.default.N400),focusText:o.default.N600,hover:o.default.N100,hoverBorder:d,hoverText:o.default.N600,text:o.default.N600,textTag:"N600"},outline:{bg:"transparent",border:r(o.default.N600),disabled:"none",disabledBorder:r(o.default.N600),dropShadow:a("darker"),focus:o.default.N200,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.N600),focusText:o.default.N600,hover:o.default.N100,hoverBorder:r(o.default.N600),hoverText:o.default.N600,text:o.default.N600,textTag:"N600"}},success:{neutral:{bg:o.default.G500,border:d,disabled:o.default.G300,disabledBorder:d,dropShadow:a("lighter"),focus:o.default.G500,focusInnerBorder:l(o.default.N700),focusOuterBorder:r(o.default.G600),focusText:o.default.N600,hover:o.default.G400,hoverBorder:a("lighter"),hoverText:o.default.N600,text:o.default.N600,textTag:"N600"}},warning:{neutral:{bg:o.default.Y500,border:d,disabled:o.default.Y300,disabledBorder:d,dropShadow:a("lighter"),focus:o.default.Y500,focusInnerBorder:l(o.default.N700),focusOuterBorder:r(o.default.Y600),focusText:o.default.N600,hover:o.default.Y400,hoverBorder:a("lighter"),hoverText:o.default.N600,text:o.default.N600,textTag:"N600"}},danger:{neutral:{bg:o.default.R500,border:d,disabled:o.default.R500,disabledBorder:d,dropShadow:a("lighter"),focus:o.default.R500,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.R600),focusText:o.default.N050,hover:o.default.R600,hoverBorder:a("lighter"),hoverText:o.default.N050,text:o.default.N050,textTag:"N050"},noFill:{bg:"transparent",border:d,disabled:"none",disabledBorder:d,dropShadow:"none",focus:"transparent",focusInnerBorder:l("transparent"),focusOuterBorder:r(o.default.R300),focusText:o.default.R500,hover:o.default.R100,hoverBorder:d,hoverText:o.default.R600,text:o.default.R500,textTag:"R500"},outline:{bg:"transparent",border:r(o.default.R500),disabled:"none",disabledBorder:r(o.default.R500),dropShadow:a("darker"),focus:o.default.R200,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.R600),focusText:o.default.R700,hover:o.default.R100,hoverBorder:r(o.default.R600),hoverText:o.default.R600,text:o.default.R500,textTag:"R500"}},default:{neutral:{bg:o.default.N500,border:d,disabled:o.default.N200,disabledBorder:r(o.default.N100),dropShadow:a("lighter"),focus:o.default.N500,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.N400),focusText:o.default.N050,hover:o.default.N300,hoverBorder:a("lighter"),hoverText:o.default.N050,text:o.default.N050,textTag:"N050"}},white:{neutral:{bg:o.default.N050,border:d,disabled:o.default.N200,disabledBorder:r(o.default.N200),dropShadow:a("lighter"),focus:o.default.N050,focusInnerBorder:l(o.default.N700),focusOuterBorder:r(o.default.N050),focusText:o.default.N500,hover:o.default.N300,hoverBorder:a("lighter"),hoverText:o.default.N500,text:o.default.N500,textTag:"N500"},noFill:{bg:"transparent",border:d,disabled:"none",disabledBorder:d,dropShadow:"none",focus:"transparent",focusInnerBorder:l("transparent"),focusOuterBorder:r(o.default.N050),focusText:o.default.N500,hover:o.default.N300,hoverBorder:a("lighter"),hoverText:o.default.N500,text:o.default.N500,textTag:"N500"}},dark:{neutral:{bg:o.default.N700,border:d,disabled:o.default.N300,disabledBorder:r(o.default.N700),dropShadow:a("darker"),focus:o.default.N700,focusInnerBorder:l(o.default.N050),focusOuterBorder:r(o.default.N700),focusText:o.default.N050,hover:o.default.N600,hoverBorder:a("darker"),hoverText:o.default.N050,text:o.default.N050,textTag:"N050"}}},c=co;var po=u.default.forwardRef((t,e)=>{let g=t,{children:n,domain:i="default",fullWidth:$,label:w,icon:p,submitting:B,variant:k="neutral"}=g,P=T(g,["children","domain","fullWidth","label","icon","submitting","variant"]),f=c[i][k]||c[i].neutral,h=p&&u.default.createElement(W,{domain:f,childrenExist:!!n},p);return u.default.createElement(R,v({"aria-label":n?void 0:w||p,domain:f,fullWidth:!!$,ref:e,type:"button"},P),u.default.createElement(O,null,u.default.createElement(I,{visible:!B},h,n),u.default.createElement(I,{visible:B},u.default.createElement(X.Submit,{color:f.textTag}))))}),C=po;var m=s(require("react")),Z=s(require("@wedgekit/indicators"));var q=s(require("styled-components")),K=s(require("@wedgekit/primitives")),fo=q.default.button`
  appearance: none;
  background: ${t=>t.buttonContext.bg};
  border: ${t=>t.buttonContext.border};
  outline: none;
  width: ${({inline:t})=>t?"1.5rem":"2.5rem"};
  height: ${({inline:t})=>t?"1.5rem":"2.5rem"};
  border-radius: 50%;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  display: flex;
  align-items: center;
  justify-content: center;

  &:disabled {
    background: ${t=>t.buttonContext.disabled};
    border: ${t=>t.buttonContext.disabledBorder};
    opacity: 0.5;
  }
  &:focus {
    border: ${t=>t.buttonContext.focusOuterBorder};
    box-shadow: ${t=>t.buttonContext.focusInnerBorder};
    background: ${t=>t.buttonContext.focus};
  }
  &:hover:not([disabled]) {
    background: ${t=>t.buttonContext.hover};
    border: ${t=>t.buttonContext.hoverBorder};
  }
  &:hover:focus {
    border: ${t=>t.buttonContext.focusOuterBorder};
    box-shadow: ${t=>t.buttonContext.focusInnerBorder};
    background: ${t=>t.buttonContext.focus};
  }

  ${K.CursorMixin}
`,D=fo;var Q=s(require("styled-components")),U=s(require("@wedgekit/icons")),bo=(0,Q.default)(U.default)`
  width: ${({inline:t})=>t?"0.75rem":"1rem"};
  min-width: ${({inline:t})=>t?"0.75rem":"1rem"};
  height: ${({inline:t})=>t?"0.75rem":"1rem"};
`,G=bo;var xo=m.default.forwardRef((t,e)=>{let E=t,{disabled:n,domain:i="secondary",icon:$,iconColor:w,inline:p,label:B,noFill:k,onClick:P,outline:f,submitting:h,variant:g="neutral"}=E,_=T(E,["disabled","domain","icon","iconColor","inline","label","noFill","onClick","outline","submitting","variant"]),N=(()=>k?c.secondary.noFill:f?c.secondary.outline:c[i][g])(),M=w||(N==null?void 0:N.textTag);return m.default.createElement(D,v({"aria-label":B,buttonContext:N,disabled:n||h,inline:p,onClick:P,ref:e,type:"button"},_),h?m.default.createElement(Z.Submit,{color:M}):m.default.createElement(G,{color:M,inline:p},$))}),S=xo;
